http-scala-fx/src/main/scala/fx/Http.scala

```diff
@@ -0,0 +1,319 @@
+package fx
+
+import java.net.URI
+import java.net.http.HttpHeaders
+import java.net.http.HttpRequest
+import java.net.http.HttpRequest.BodyPublishers
+import java.net.http.HttpResponse
+import java.net.http.HttpResponse.BodyHandlers
+import java.time.Duration
+import java.util.Random
+import scala.annotation.tailrec
+import scala.concurrent.duration.{Duration => ScalaDuration}
+
+/**
+ * Models the http scheme
+ */
+type Http[A] = (
+    HttpRetryPolicy[A],
+    Control[HttpExecutionException],
+    HttpClient
+) ?=> A
+
+extension [A](http: Http[A])
+  def httpValue(using Control[HttpExecutionException]): A = http
+  def fmap[B](f: A => B): Http[B] =
```

Comment:
why not `map` and `flatMap`? I think if they are useful, they might as well get nicer syntax for `for`

Reply:
Due to `type Http[A]` not being `opaque type Http[A]`, this collides with any other top-level `map`/`flatMap` definition. The type needs to be `opaque`, but i couldn't make the `monaderror` in sttp recognize the context parameters.

Reply:
Basically, I'm running into compiler bugs with the opaque Http type in sttp's MonadError:

```scala
[info] compiling 8 Scala sources to /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/target/scala-3.1.2/classes ...
exception while typing def flatMap[T, T2](fa: fx.Http[T])(f: T => fx.Http[T2]): fx.Http[T2] = fx.Http.flatMap[T](fa)[T2](f) of class class dotty.tools.dotc.ast.Trees$DefDef # -1
exception while typing final class $anon() extends Object(), sttp.monad.MonadError[fx.Http] {
  def ensure[T](f: fx.Http[T], e: () => fx.Http[Unit]): fx.Http[T] = 
    {
      val x: Throwable | T = 
        fx.run[Exception, T](
          {
            def $anonfun(using evidence$3: fx.Control[Exception]): T = fx.Http.value[T](f)(evidence$3)
            closure($anonfun)
          }
        )
      matchResult2[Function3]: 
        {
          case val x14: (x : Object) = x
          if x14.$isInstanceOf[Throwable] then 
            {
              case val ex: Throwable = x14.$asInstanceOf[Throwable]
              return[matchResult2] 
                {
                  val ignored: fx.Http[Unit] = 
                    fx.run[Nothing, fx.Http[Unit]](
                      {
                        def $anonfun(using evidence$4: fx.Control[Nothing]): fx.Http[Unit] = e.apply()
                        closure($anonfun)
                      }
                    )
                  f
                }
            }
           else ()
          case val a: Throwable | T = x14
          return[matchResult2] 
            {
              this.unit[T](a.asInstanceOf[T])
            }
        }
    }
  def error[T](t: Throwable): fx.Http[T] = sttp.fx.HttpScalaFXBackend.this.control.shift(t)[Nothing]
  def flatMap[T, T2](fa: fx.Http[T])(f: T => fx.Http[T2]): fx.Http[T2] = fx.Http.flatMap[T](fa)[T2](f)
  protected def handleWrappedError[T](rt: fx.Http[T])(h: PartialFunction[Throwable, fx.Http[T]]): fx.Http[T] = 
    {
      val x: Throwable | T = 
        fx.run[Exception, T](
          {
            def $anonfun(using evidence$5: fx.Control[Exception]): T = fx.Http.value[T](rt)(evidence$5)
            closure($anonfun)
          }
        )
      matchResult3[Function3]: 
        {
          case val x16: (x : Object) = x
          if x16.$isInstanceOf[Throwable] then 
            {
              case val x18: Throwable = x16.$asInstanceOf[Throwable]
              {
                case val ex: Throwable = x18
                if h.isDefinedAt(ex) then 
                  return[matchResult3] 
                    {
                      h.apply(ex)
                    }
                 else ()
              }
              case val ex: Throwable = x18
              return[matchResult3] 
                {
                  fx.raise[fx.HttpExecutionException](fx.HttpExecutionException.apply(new RuntimeException(ex)))[Nothing](
                    sttp.fx.HttpScalaFXBackend.this.control
                  )
                }
            }
           else ()
          case val a: Throwable | T = x16
          return[matchResult3] 
            {
              this.unit[T](a.asInstanceOf[T])
            }
        }
    }
  def map[T, T2](fa: fx.Http[T])(f: T => T2): fx.Http[T2] = fx.Http.map[T](fa)[T2](f)
  def unit[T](t: T): fx.Http[T] = fx.Http.apply[T](t)
  private val $outer: sttp.fx.HttpScalaFXBackend
  final def sttp$fx$HttpScalaFXBackend$_$$anon$$$outer: sttp.fx.HttpScalaFXBackend = $outer
} of class class dotty.tools.dotc.ast.Trees$TypeDef # -1
exception while typing {
  final class $anon() extends Object(), sttp.monad.MonadError[fx.Http] {
    def ensure[T](f: fx.Http[T], e: () => fx.Http[Unit]): fx.Http[T] = 
      {
        val x: Throwable | T = 
          fx.run[Exception, T](
            {
              def $anonfun(using evidence$3: fx.Control[Exception]): T = fx.Http.value[T](f)(evidence$3)
              closure($anonfun)
            }
          )
        matchResult2[Function3]: 
          {
            case val x14: (x : Object) = x
            if x14.$isInstanceOf[Throwable] then 
              {
                case val ex: Throwable = x14.$asInstanceOf[Throwable]
                return[matchResult2] 
                  {
                    val ignored: fx.Http[Unit] = 
                      fx.run[Nothing, fx.Http[Unit]](
                        {
                          def $anonfun(using evidence$4: fx.Control[Nothing]): fx.Http[Unit] = e.apply()
                          closure($anonfun)
                        }
                      )
                    f
                  }
              }
             else ()
            case val a: Throwable | T = x14
            return[matchResult2] 
              {
                this.unit[T](a.asInstanceOf[T])
              }
          }
      }
    def error[T](t: Throwable): fx.Http[T] = sttp.fx.HttpScalaFXBackend.this.control.shift(t)[Nothing]
    def flatMap[T, T2](fa: fx.Http[T])(f: T => fx.Http[T2]): fx.Http[T2] = fx.Http.flatMap[T](fa)[T2](f)
    protected def handleWrappedError[T](rt: fx.Http[T])(h: PartialFunction[Throwable, fx.Http[T]]): fx.Http[T] = 
      {
        val x: Throwable | T = 
          fx.run[Exception, T](
            {
              def $anonfun(using evidence$5: fx.Control[Exception]): T = fx.Http.value[T](rt)(evidence$5)
              closure($anonfun)
            }
          )
        matchResult3[Function3]: 
          {
            case val x16: (x : Object) = x
            if x16.$isInstanceOf[Throwable] then 
              {
                case val x18: Throwable = x16.$asInstanceOf[Throwable]
                {
                  case val ex: Throwable = x18
                  if h.isDefinedAt(ex) then 
                    return[matchResult3] 
                      {
                        h.apply(ex)
                      }
                   else ()
                }
                case val ex: Throwable = x18
                return[matchResult3] 
                  {
                    fx.raise[fx.HttpExecutionException](fx.HttpExecutionException.apply(new RuntimeException(ex)))[Nothing](
                      sttp.fx.HttpScalaFXBackend.this.control
                    )
                  }
              }
             else ()
            case val a: Throwable | T = x16
            return[matchResult3] 
              {
                this.unit[T](a.asInstanceOf[T])
              }
          }
      }
    def map[T, T2](fa: fx.Http[T])(f: T => T2): fx.Http[T2] = fx.Http.map[T](fa)[T2](f)
    def unit[T](t: T): fx.Http[T] = fx.Http.apply[T](t)
    private val $outer: sttp.fx.HttpScalaFXBackend
    final def sttp$fx$HttpScalaFXBackend$_$$anon$$$outer: sttp.fx.HttpScalaFXBackend = $outer
  }
  new Object with sttp.monad.MonadError {...}():sttp.monad.MonadError[fx.Http]
} of class class dotty.tools.dotc.ast.Trees$Block # -1
exception while typing def responseMonad: sttp.monad.MonadError[fx.Http] = 
  {
    final class $anon() extends Object(), sttp.monad.MonadError[fx.Http] {
      def ensure[T](f: fx.Http[T], e: () => fx.Http[Unit]): fx.Http[T] = 
        {
          val x: Throwable | T = 
            fx.run[Exception, T](
              {
                def $anonfun(using evidence$3: fx.Control[Exception]): T = fx.Http.value[T](f)(evidence$3)
                closure($anonfun)
              }
            )
          matchResult2[Function3]: 
            {
              case val x14: (x : Object) = x
              if x14.$isInstanceOf[Throwable] then 
                {
                  case val ex: Throwable = x14.$asInstanceOf[Throwable]
                  return[matchResult2] 
                    {
                      val ignored: fx.Http[Unit] = 
                        fx.run[Nothing, fx.Http[Unit]](
                          {
                            def $anonfun(using evidence$4: fx.Control[Nothing]): fx.Http[Unit] = e.apply()
                            closure($anonfun)
                          }
                        )
                      f
                    }
                }
               else ()
              case val a: Throwable | T = x14
              return[matchResult2] 
                {
                  this.unit[T](a.asInstanceOf[T])
                }
            }
        }
      def error[T](t: Throwable): fx.Http[T] = sttp.fx.HttpScalaFXBackend.this.control.shift(t)[Nothing]
      def flatMap[T, T2](fa: fx.Http[T])(f: T => fx.Http[T2]): fx.Http[T2] = fx.Http.flatMap[T](fa)[T2](f)
      protected def handleWrappedError[T](rt: fx.Http[T])(h: PartialFunction[Throwable, fx.Http[T]]): fx.Http[T] = 
        {
          val x: Throwable | T = 
            fx.run[Exception, T](
              {
                def $anonfun(using evidence$5: fx.Control[Exception]): T = fx.Http.value[T](rt)(evidence$5)
                closure($anonfun)
              }
            )
          matchResult3[Function3]: 
            {
              case val x16: (x : Object) = x
              if x16.$isInstanceOf[Throwable] then 
                {
                  case val x18: Throwable = x16.$asInstanceOf[Throwable]
                  {
                    case val ex: Throwable = x18
                    if h.isDefinedAt(ex) then 
                      return[matchResult3] 
                        {
                          h.apply(ex)
                        }
                     else ()
                  }
                  case val ex: Throwable = x18
                  return[matchResult3] 
                    {
                      fx.raise[fx.HttpExecutionException](fx.HttpExecutionException.apply(new RuntimeException(ex)))[Nothing](
                        sttp.fx.HttpScalaFXBackend.this.control
                      )
                    }
                }
               else ()
              case val a: Throwable | T = x16
              return[matchResult3] 
                {
                  this.unit[T](a.asInstanceOf[T])
                }
            }
        }
      def map[T, T2](fa: fx.Http[T])(f: T => T2): fx.Http[T2] = fx.Http.map[T](fa)[T2](f)
      def unit[T](t: T): fx.Http[T] = fx.Http.apply[T](t)
      private val $outer: sttp.fx.HttpScalaFXBackend
      final def sttp$fx$HttpScalaFXBackend$_$$anon$$$outer: sttp.fx.HttpScalaFXBackend = $outer
    }
    new Object with sttp.monad.MonadError {...}():sttp.monad.MonadError[fx.Http]
  } of class class dotty.tools.dotc.ast.Trees$DefDef # -1
exception while typing @SourceFile("sttp-scala-fx/src/main/scala/sttp/fx/HttpScalaFXBackend.scala") class HttpScalaFXBackend()(using client: fx.HttpClient, 
  control: fx.Control[Throwable | fx.HttpExecutionException]
, config: fx.HttpClientConfig) extends Object(), sttp.client3.SttpBackend[fx.Http, sttp.fx.ReceiveStreams] {
  private given val client: fx.HttpClient
  private given val control: fx.Control[Throwable | fx.HttpExecutionException]
  private given val config: fx.HttpClientConfig
  private final lazy given def given_MonadError_Http: sttp.monad.MonadError[fx.Http] = this.responseMonad
  def send[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]](request: sttp.client3.Request[T, R]): fx.Http[sttp.client3.Response[T]] = 
    matchResult1[Function3]: 
      {
        case val x1: (request.body : (): sttp.client3.RequestBody) = request.body
        if x1.eq(sttp.client3.NoBody) then 
          {
            case val x: sttp.client3.NoBody.type = x1.$asInstanceOf[sttp.client3.NoBody.type]
            return[matchResult1] 
              {
                this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.NoBody.type](request, x)(
                  sttp.fx.given_ToHttpBodyMapper_NoBody_type
                , fx.HttpResponseMapper.HttpResponseReceiveMapper)
              }
          }
         else ()
        if x1.$isInstanceOf[sttp.client3.StringBody] then 
          {
            case val x: sttp.client3.StringBody = x1.$asInstanceOf[sttp.client3.StringBody]
            return[matchResult1] 
              {
                this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.StringBody](request, x)(
                  sttp.fx.given_ToHttpBodyMapper_StringBody
                , fx.HttpResponseMapper.HttpResponseReceiveMapper)
              }
          }
         else ()
        if x1.$isInstanceOf[sttp.client3.ByteArrayBody] then 
          {
            case val x: sttp.client3.ByteArrayBody = x1.$asInstanceOf[sttp.client3.ByteArrayBody]
            return[matchResult1] 
              {
                this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.ByteArrayBody](request, x)(
                  sttp.fx.given_ToHttpBodyMapper_ByteArrayBody
                , fx.HttpResponseMapper.HttpResponseReceiveMapper)
              }
          }
         else ()
        if x1.$isInstanceOf[sttp.client3.ByteBufferBody] then 
          {
            case val x: sttp.client3.ByteBufferBody = x1.$asInstanceOf[sttp.client3.ByteBufferBody]
            return[matchResult1] 
              {
                this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.ByteBufferBody](request, x)(
                  sttp.fx.given_ToHttpBodyMapper_ByteBufferBody
                , fx.HttpResponseMapper.HttpResponseReceiveMapper)
              }
          }
         else ()
        if x1.$isInstanceOf[sttp.client3.InputStreamBody] then 
          {
            case val x: sttp.client3.InputStreamBody = x1.$asInstanceOf[sttp.client3.InputStreamBody]
            return[matchResult1] 
              {
                this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.InputStreamBody](request, x)(
                  sttp.fx.given_ToHttpBodyMapper_InputStreamBody
                , fx.HttpResponseMapper.HttpResponseReceiveMapper)
              }
          }
         else ()
        if x1.$isInstanceOf[sttp.client3.FileBody] then 
          {
            case val x: sttp.client3.FileBody = x1.$asInstanceOf[sttp.client3.FileBody]
            return[matchResult1] 
              {
                this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.FileBody](request, x)(
                  sttp.fx.given_ToHttpBodyMapper_FileBody
                , fx.HttpResponseMapper.HttpResponseReceiveMapper)
              }
          }
         else ()
        if x1.$isInstanceOf[sttp.client3.StreamBody[BinaryStream$1, S$1]] then 
          {
            case val x6: sttp.client3.StreamBody[BinaryStream$1, S$1] = 
              sttp.client3.StreamBody.unapply[BinaryStream$1 @ BinaryStream$1, S$1 @ S$1](x1.$asInstanceOf[sttp.client3.StreamBody[BinaryStream$1, S$1]])
            case val x7: Any = x6._1
            case val x: sttp.client3.StreamBody[BinaryStream$1, S$1] = x1.$asInstanceOf[sttp.client3.StreamBody[BinaryStream$1, S$1]]
            return[matchResult1] 
              {
                val body: sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams] = 
                  {
                    val $proxy1: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                      fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                    (try 
                      {
                        {
                          def $anonfun(using evidence$1: fx.Throws[Exception]): sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams] = 
                            x.asInstanceOf[sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]]
                          closure($anonfun)
                        }.$asInstanceOf[
                          ((fx.Throws[Exception]) ?=> sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]) & (($proxy1.Throws[Exception]) ?=>
                             
                          sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams])
                        ].apply(fx.Throws.unsafeExceptions[Exception])
                      }
                     catch 
                      {
                        case r @ _:Exception => 
                          {
                            HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[
                              sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]
                            ]
                          }
                      }
                    ):sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]
                  }
                this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), 
                  sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]
                ](request, body)(sttp.fx.given_ToHttpBodyMapper_StreamBody, fx.HttpResponseMapper.HttpResponseReceiveMapper)
              }
          }
         else ()
        if x1.$isInstanceOf[sttp.client3.MultipartBody[R$1]] then 
          {
            case val x3: sttp.client3.MultipartBody[R$1] = sttp.client3.MultipartBody.unapply[R$1 @ R$1](x1.$asInstanceOf[sttp.client3.MultipartBody[R$1]])
            case val x4: scala.collection.immutable.Seq[sttp.model.Part[sttp.client3.RequestBody[R$1]]] = x3._1
            case val x: sttp.client3.MultipartBody[R$1] = x1.$asInstanceOf[sttp.client3.MultipartBody[R$1]]
            return[matchResult1] 
              {
                val body: sttp.client3.MultipartBody[R] = 
                  {
                    val $proxy2: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                      fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                    (try 
                      {
                        {
                          def $anonfun(using evidence$2: fx.Throws[Exception]): sttp.client3.MultipartBody[R] = 
                            x.asInstanceOf[sttp.client3.MultipartBody[R]]
                          closure($anonfun)
                        }.$asInstanceOf[
                          ((fx.Throws[Exception]) ?=> sttp.client3.MultipartBody[R]) & (($proxy2.Throws[Exception]) ?=> sttp.client3.MultipartBody[R])
                        ].apply(fx.Throws.unsafeExceptions[Exception])
                      }
                     catch 
                      {
                        case r @ _:Exception => 
                          {
                            HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[sttp.client3.MultipartBody[R]]
                          }
                      }
                    ):sttp.client3.MultipartBody[R]
                  }
                this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.MultipartBody[R]](request, body)(
                  sttp.fx.multipartBodyMapper[R]
                , fx.HttpResponseMapper.HttpResponseReceiveMapper)
              }
          }
         else ()
        throw new MatchError(x1)
      }
  def close(): fx.Http[Unit] = ???
  def responseMonad: sttp.monad.MonadError[fx.Http] = 
    {
      final class $anon() extends Object(), sttp.monad.MonadError[fx.Http] {
        def ensure[T](f: fx.Http[T], e: () => fx.Http[Unit]): fx.Http[T] = 
          {
            val x: Throwable | T = 
              fx.run[Exception, T](
                {
                  def $anonfun(using evidence$3: fx.Control[Exception]): T = fx.Http.value[T](f)(evidence$3)
                  closure($anonfun)
                }
              )
            matchResult2[Function3]: 
              {
                case val x14: (x : Object) = x
                if x14.$isInstanceOf[Throwable] then 
                  {
                    case val ex: Throwable = x14.$asInstanceOf[Throwable]
                    return[matchResult2] 
                      {
                        val ignored: fx.Http[Unit] = 
                          fx.run[Nothing, fx.Http[Unit]](
                            {
                              def $anonfun(using evidence$4: fx.Control[Nothing]): fx.Http[Unit] = e.apply()
                              closure($anonfun)
                            }
                          )
                        f
                      }
                  }
                 else ()
                case val a: Throwable | T = x14
                return[matchResult2] 
                  {
                    this.unit[T](a.asInstanceOf[T])
                  }
              }
          }
        def error[T](t: Throwable): fx.Http[T] = sttp.fx.HttpScalaFXBackend.this.control.shift(t)[Nothing]
        def flatMap[T, T2](fa: fx.Http[T])(f: T => fx.Http[T2]): fx.Http[T2] = fx.Http.flatMap[T](fa)[T2](f)
        protected def handleWrappedError[T](rt: fx.Http[T])(h: PartialFunction[Throwable, fx.Http[T]]): fx.Http[T] = 
          {
            val x: Throwable | T = 
              fx.run[Exception, T](
                {
                  def $anonfun(using evidence$5: fx.Control[Exception]): T = fx.Http.value[T](rt)(evidence$5)
                  closure($anonfun)
                }
              )
            matchResult3[Function3]: 
              {
                case val x16: (x : Object) = x
                if x16.$isInstanceOf[Throwable] then 
                  {
                    case val x18: Throwable = x16.$asInstanceOf[Throwable]
                    {
                      case val ex: Throwable = x18
                      if h.isDefinedAt(ex) then 
                        return[matchResult3] 
                          {
                            h.apply(ex)
                          }
                       else ()
                    }
                    case val ex: Throwable = x18
                    return[matchResult3] 
                      {
                        fx.raise[fx.HttpExecutionException](fx.HttpExecutionException.apply(new RuntimeException(ex)))[Nothing](
                          sttp.fx.HttpScalaFXBackend.this.control
                        )
                      }
                  }
                 else ()
                case val a: Throwable | T = x16
                return[matchResult3] 
                  {
                    this.unit[T](a.asInstanceOf[T])
                  }
              }
          }
        def map[T, T2](fa: fx.Http[T])(f: T => T2): fx.Http[T2] = fx.Http.map[T](fa)[T2](f)
        def unit[T](t: T): fx.Http[T] = fx.Http.apply[T](t)
        private val $outer: sttp.fx.HttpScalaFXBackend
        final def sttp$fx$HttpScalaFXBackend$_$$anon$$$outer: sttp.fx.HttpScalaFXBackend = $outer
      }
      new Object with sttp.monad.MonadError {...}():sttp.monad.MonadError[fx.Http]
    }
  private def requestAsBody[R, A <: sttp.client3.RequestBody[R]](r: A)(using x$2: sttp.fx.ToHttpBodyMapper[A]): fx.Http[fx.HttpBodyMapper[A]] = 
    {
      val $proxy3: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
        fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
      (try 
        {
          {
            def $anonfun(using evidence$6: fx.Throws[Exception]): fx.Http[fx.HttpBodyMapper[A]] = 
              fx.Http.apply[fx.HttpBodyMapper[A]](x$2.toHttpBodyMapper(r)())
            closure($anonfun)
          }.$asInstanceOf[((fx.Throws[Exception]) ?=> fx.Http[fx.HttpBodyMapper[A]]) & (($proxy3.Throws[Exception]) ?=> fx.Http[fx.HttpBodyMapper[A]])].
            apply
          (fx.Throws.unsafeExceptions[Exception])
        }
       catch 
        {
          case r @ _:Exception => 
            {
              def $anonfun(e: Exception): fx.Http[fx.HttpBodyMapper[A]] = 
                HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(e))[Nothing]
              closure($anonfun)
            }.$asInstanceOf[Exception => fx.Http[fx.HttpBodyMapper[A]]].apply(r)
        }
      ):fx.Http[fx.HttpBodyMapper[A]]
    }
  private def getRequestHeaders[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http], A](request: sttp.client3.Request[T, R], body: A)(using 
    bodyMapper: fx.HttpBodyMapper[A]
  ): fx.Http[scala.collection.immutable.::[fx.HttpHeader]] = 
    {
      val Content-Type: fx.HttpHeader = 
        fx.HttpHeader.apply(
          {
            val ev$1: ArrowAssoc[String] = ArrowAssoc[String]("Content-Type")
            scala.Predef.ArrowAssoc.->$extension[scala.collection.immutable.::[String], String](ev$1)(
              ::.apply[String](fx.MediaType.value(bodyMapper.mediaType), List.empty[String])
            )
          }
        )
      fx.Http.apply[scala.collection.immutable.::[fx.HttpHeader]](
        ::.apply[(String, scala.collection.immutable.::[String])](`Content-Type`, 
          request.headers.filterNot(
            {
              def $anonfun(_$1: sttp.model.Header): Boolean = _$1.name.==("Content-Length").unary_!
              closure($anonfun)
            }
          ).groupBy[String](
            {
              def $anonfun(_$2: sttp.model.Header): String = _$2.name
              closure($anonfun)
            }
          ).map[String, scala.collection.immutable.::[String]](
            {
              {
                def $anonfun(p: (String, scala.collection.immutable.Seq[sttp.model.Header])): fx.HttpHeader = 
                  {
                    val values: scala.collection.immutable.List[String] = 
                      p._2.map[String](
                        {
                          def $anonfun(_$3: sttp.model.Header): String = _$3.value
                          closure($anonfun)
                        }
                      ).toList
                    fx.HttpHeader.apply(p._1, 
                      values.headOption.getOrElse[String](
                        <special-ops>.<cbn-arg>[("" : String)](
                          {
                            def $anonfun(): ("" : String) = ""
                            closure($anonfun)
                          }
                        )
                      )
                    , values.tail)
                  }
                closure($anonfun)
              }
            }
          ).toList
        )
      )
    }
  private def getResponseHeaders[T](response: java.net.http.HttpResponse[T]): fx.Http[scala.collection.immutable.Seq[sttp.model.Header]] = 
    {
      fx.Http.apply[scala.collection.immutable.Seq[sttp.model.Header]](
        jdk.CollectionConverters.MapHasAsScala[String, java.util.List[String]](response.headers().map()).asScala.toList.foldLeft[
          scala.collection.immutable.Seq[sttp.model.Header]
        ](Seq.empty[sttp.model.Header])(
          {
            {
              def $anonfun(acc: scala.collection.immutable.Seq[sttp.model.Header], jnhHeader: (String, java.util.List[String])): 
                scala.collection.immutable.Seq[sttp.model.Header]
               = 
                {
                  acc.++[sttp.model.Header](
                    jdk.CollectionConverters.ListHasAsScala[String](jnhHeader._2).asScala.map[sttp.model.Header](
                      {
                        def $anonfun(_$4: String): sttp.model.Header = sttp.model.Header.apply(jnhHeader._1, _$4)
                        closure($anonfun)
                      }
                    )
                  )
                }
              closure($anonfun)
            }
          }
        )
      )
    }
  private def bodyAsResponseAs[T](using x$1: sttp.monad.MonadError[fx.Http]): 
    sttp.client3.internal.BodyFromResponseAs[fx.Http, java.net.http.HttpResponse[fx.Receive[Byte]], Nothing, fx.Receive[Byte]]
   = 
    {
      final class $anon() extends 
        sttp.client3.internal.BodyFromResponseAs[fx.Http, java.net.http.HttpResponse[fx.Receive[Byte]], Nothing, fx.Receive[Byte]]
      ()(x$1) {
        protected override def withReplayableBody(response: java.net.http.HttpResponse[fx.Receive[Byte]], 
          replayableBody: scala.util.Either[Array[Byte], sttp.client3.internal.SttpFile]
        ): fx.Http[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
          fx.Http.apply[java.net.http.HttpResponse[fx.Receive[Byte]]](
            {
              final class $anon() extends Object(), java.net.http.HttpResponse[fx.Receive[Byte]] {
                def statusCode(): Int = response.statusCode()
                def request(): java.net.http.HttpRequest = response.request()
                def previousResponse(): java.util.Optional[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
                  java.util.Optional.of[java.net.http.HttpResponse[fx.Receive[Byte]]](response)
                def headers(): java.net.http.HttpHeaders = response.headers()
                def body(): fx.Receive[Byte] = 
                  replayableBody.fold[fx.Receive[Byte]](
                    {
                      def $anonfun(bytes: Array[Byte]): fx.Receive[Byte] = fx.streamOf[Byte](wrapByteArray(bytes).toList)
                      closure($anonfun)
                    }
                  , 
                    {
                      def $anonfun(file: sttp.client3.internal.SttpFile): fx.Receive[Byte] = fx.streamOf[Byte](wrapByteArray(file.readAsByteArray).toList)
                      closure($anonfun)
                    }
                  )
                def sslSession(): java.util.Optional[javax.net.ssl.SSLSession] = response.sslSession()
                def uri(): java.net.URI = response.uri()
                def version(): java.net.http.HttpClient.Version = response.version()
              }
              new Object with java.net.http.HttpResponse {...}():java.net.http.HttpResponse[fx.Receive[Byte]]
            }
          )
        protected override def regularIgnore(response: java.net.http.HttpResponse[fx.Receive[Byte]]): fx.Http[Unit] = 
          fx.Http.map[fx.Receive[Byte]](
            fx.Http.apply[fx.Receive[Byte]](
              {
                val $proxy4: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                  fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                {
                  def liftedTree1(): fx.Receive[Byte] = 
                    (try 
                      {
                        {
                          def $anonfun(using evidence$7: fx.Throws[Exception]): fx.Receive[Byte] = response.body()
                          closure($anonfun)
                        }.$asInstanceOf[((fx.Throws[Exception]) ?=> fx.Receive[Byte]) & (($proxy4.Throws[Exception]) ?=> fx.Receive[Byte])].apply(
                          fx.Throws.unsafeExceptions[Exception]
                        )
                      }
                     catch 
                      {
                        case r @ _:Exception => 
                          {
                            sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[fx.Receive[Byte]]
                          }
                      }
                    )
                  liftedTree1()
                }:fx.Receive[Byte]
              }
            )
          )[Unit](
            {
              def $anonfun(x$1: fx.Receive[Byte]): Unit = 
                matchResult4[Unit]: 
                  {
                    case val x19: (x$1 : fx.Receive) @unchecked = x$1:(x$1 : fx.Receive) @unchecked
                    return[matchResult4] 
                      {
                        fx.Http.apply[Unit](())
                        ()
                      }
                  }
              closure($anonfun)
            }
          )
        protected override def regularAsByteArray(response: java.net.http.HttpResponse[fx.Receive[Byte]]): fx.Http[Array[Byte]] = 
          fx.Http.apply[Array[Byte]](
            {
              val $proxy5: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
              {
                def liftedTree2(): Array[Byte] = 
                  (try 
                    {
                      {
                        def $anonfun(using evidence$8: fx.Throws[Exception]): Array[Byte] = 
                          fx.toList[Byte](response.body()).toArray[Byte](scala.reflect.ClassTag.apply[Byte](classOf[Byte]))
                        closure($anonfun)
                      }.$asInstanceOf[((fx.Throws[Exception]) ?=> Array[Byte]) & (($proxy5.Throws[Exception]) ?=> Array[Byte])].apply(
                        fx.Throws.unsafeExceptions[Exception]
                      )
                    }
                   catch 
                    {
                      case r @ _:Exception => 
                        {
                          sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[Array[Byte]]
                        }
                    }
                  )
                liftedTree2()
              }:Array[Byte]
            }
          )
        protected override def regularAsFile(response: java.net.http.HttpResponse[fx.Receive[Byte]], file: sttp.client3.internal.SttpFile): 
          fx.Http[sttp.client3.internal.SttpFile]
         = 
          fx.Http.map[java.nio.file.Path](
            fx.Http.apply[java.nio.file.Path](
              {
                val $proxy6: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                  fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                {
                  def liftedTree3(): java.nio.file.Path = 
                    (try 
                      {
                        {
                          def $anonfun(using evidence$9: fx.Throws[Exception]): java.nio.file.Path = 
                            java.nio.file.Files.write(file.toPath, 
                              fx.toList[Byte](response.body()).toArray[Byte](scala.reflect.ClassTag.apply[Byte](classOf[Byte]))
                            , [ : java.nio.file.OpenOption])
                          closure($anonfun)
                        }.$asInstanceOf[((fx.Throws[Exception]) ?=> java.nio.file.Path) & (($proxy6.Throws[Exception]) ?=> java.nio.file.Path)].apply(
                          fx.Throws.unsafeExceptions[Exception]
                        )
                      }
                     catch 
                      {
                        case r @ _:Exception => 
                          {
                            sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[java.nio.file.Path]
                          }
                      }
                    )
                  liftedTree3()
                }:java.nio.file.Path
              }
            )
          )[sttp.client3.internal.SttpFile](
            {
              def $anonfun(x$1: java.nio.file.Path): sttp.client3.internal.SttpFile = 
                matchResult5[sttp.client3.internal.SttpFile]: 
                  {
                    case val x20: (x$1 : java.nio.file.Path) @unchecked = x$1:(x$1 : java.nio.file.Path) @unchecked
                    return[matchResult5] file
                  }
              closure($anonfun)
            }
          )
        protected override def regularAsStream(response: java.net.http.HttpResponse[fx.Receive[Byte]]): fx.Http[(fx.Receive[Byte], () => fx.Http[Unit])] = 
          fx.Http.apply[(fx.Receive[Byte], () => fx.Http[Unit])](
            {
              val $proxy7: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
              {
                def liftedTree4(): (fx.Receive[Byte], () => fx.Http[Unit]) = 
                  (try 
                    {
                      {
                        def $anonfun(using evidence$10: fx.Throws[Exception]): (fx.Receive[Byte], () => fx.Http[Unit]) = 
                          Tuple2.apply[fx.Receive[Byte], () => fx.Http[Unit]](response.body(), 
                            {
                              def $anonfun(): fx.Http[Unit] = fx.Http.apply[Unit](())
                              closure($anonfun)
                            }
                          )
                        closure($anonfun)
                      }.$asInstanceOf[
                        ((fx.Throws[Exception]) ?=> (fx.Receive[Byte], () => fx.Http[Unit])) & (($proxy7.Throws[Exception]) ?=> (fx.Receive[Byte], () => 
                          fx.Http[Unit]
                        ))
                      ].apply(fx.Throws.unsafeExceptions[Exception])
                    }
                   catch 
                    {
                      case r @ _:Exception => 
                        {
                          sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[(fx.Receive[Byte], () => fx.Http[Unit])]
                        }
                    }
                  )
                liftedTree4()
              }:(fx.Receive[Byte], () => fx.Http[Unit])
            }
          )
        protected override def handleWS[T](responseAs: sttp.client3.WebSocketResponseAs[T, ?], meta: sttp.model.ResponseMetadata, ws: Nothing): fx.Http[T]
           = 
        fx.Http.apply[T](
          sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(new RuntimeException("Websockets are unsupported")))[T]
        )
        protected override def cleanupWhenNotAWebSocket(response: java.net.http.HttpResponse[fx.Receive[Byte]], e: sttp.client3.ws.NotAWebSocketException)
          : 
        fx.Http[Unit] = fx.Http.apply[Unit](())
        protected override def cleanupWhenGotWebSocket(response: Nothing, e: sttp.client3.ws.GotAWebSocketException): fx.Http[Unit] = 
          fx.Http.apply[Unit](
            sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(new RuntimeException("Websockets are unsupported")))[Unit]
          )
        private val $outer: sttp.fx.HttpScalaFXBackend
        final def sttp$fx$HttpScalaFXBackend$_$$anon$$$outer: sttp.fx.HttpScalaFXBackend = $outer
      }
      new sttp.client3.internal.BodyFromResponseAs {...}():
        sttp.client3.internal.BodyFromResponseAs[fx.Http, java.net.http.HttpResponse[fx.Receive[Byte]], Nothing, fx.Receive[Byte]]
    }
  private def toResponse[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]](request: sttp.client3.Request[T, R], 
    response: java.net.http.HttpResponse[fx.Receive[Byte]]
  )(using x$3: sttp.fx.StatusCodeToStatusCode[sttp.model.StatusCode, fx.StatusCode], x$4: sttp.fx.StatusCodeToStatusCode[Int, sttp.model.StatusCode]): 
    fx.Http[sttp.client3.Response[T]]
   = 
    {
      val status: sttp.model.StatusCode = 
        fx.Http.value[sttp.model.StatusCode](x$4.toStatusCode(response.statusCode())()(HttpScalaFXBackend.this.control))(HttpScalaFXBackend.this.control)
      val statusText: String = 
        fx.StatusCode.statusText(fx.Http.value[fx.StatusCode](x$3.toStatusCode(status)()(HttpScalaFXBackend.this.control))(HttpScalaFXBackend.this.control)
          )
      val headers: scala.collection.immutable.Seq[sttp.model.Header] = 
        fx.Http.value[scala.collection.immutable.Seq[sttp.model.Header]](this.getResponseHeaders[fx.Receive[Byte]](response))(
          HttpScalaFXBackend.this.control
        )
      val responseMetadata: sttp.model.ResponseMetadata = sttp.model.ResponseMetadata.apply(status, statusText, headers)
      val body: T = 
        fx.Http.value[T](
          this.bodyAsResponseAs[T](this.given_MonadError_Http).apply[T](request.response, responseMetadata, 
            Left.apply[java.net.http.HttpResponse[fx.Receive[Byte]], Nothing](response)
          )
        )(HttpScalaFXBackend.this.control)
      fx.Http.apply[sttp.client3.Response[T]](sttp.client3.Response.apply[T](body, status, statusText, headers))
    }
  private def getUri[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]](request: sttp.client3.Request[T, R]): fx.Http[java.net.URI] = 
    {
      fx.Http.apply[java.net.URI](
        {
          val $proxy8: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
            fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
          {
            def liftedTree5(): java.net.URI = 
              (try 
                {
                  {
                    def $anonfun(using evidence$11: fx.Throws[Exception]): java.net.URI = request.uri.toJavaUri
                    closure($anonfun)
                  }.$asInstanceOf[((fx.Throws[Exception]) ?=> java.net.URI) & (($proxy8.Throws[Exception]) ?=> java.net.URI)].apply(
                    fx.Throws.unsafeExceptions[Exception]
                  )
                }
               catch 
                {
                  case r @ _:Exception => 
                    {
                      def $anonfun(e: Exception): java.net.URI = HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(e))[Nothing]
                      closure($anonfun)
                    }.$asInstanceOf[Exception => java.net.URI].apply(r)
                }
              )
            liftedTree5()
          }:java.net.URI
        }
      )
    }
  private def makeRequest[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http], A](request: sttp.client3.Request[T, R], body: A)(using 
    x$3: sttp.fx.ToHttpBodyMapper[A]
  , x$4: fx.HttpResponseMapper[fx.Receive[Byte]]): fx.Http[sttp.client3.Response[T]] = 
    {
      final lazy given def bm: fx.HttpBodyMapper[A] = x$3.toHttpBodyMapper(body)()
      val headers: scala.collection.immutable.::[fx.HttpHeader] = 
        fx.Http.value[scala.collection.immutable.::[fx.HttpHeader]](
          this.getRequestHeaders[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), A](request, body)(bm)
        )(HttpScalaFXBackend.this.control)
      val uri: java.net.URI = 
        fx.Http.value[java.net.URI](this.getUri[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request))(HttpScalaFXBackend.this.control)
      val timeout: scala.concurrent.duration.Duration = request.options.readTimeout
      matchResult6[Function3]: 
        {
          case val x21: (request.method : (): Object) = request.method
          if sttp.model.Method.DELETE.==(x21) then 
            return[matchResult6] 
              {
                this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                  fx.DELETE(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                    HttpScalaFXBackend.this.control
                  , HttpScalaFXBackend.this.client)
                )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
              }
           else ()
          if sttp.model.Method.GET.==(x21) then 
            return[matchResult6] 
              {
                this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                  fx.GET(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                    HttpScalaFXBackend.this.control
                  , HttpScalaFXBackend.this.client)
                )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
              }
           else ()
          if sttp.model.Method.HEAD.==(x21) then 
            return[matchResult6] 
              {
                val response: fx.Http[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
                  fx.Http.map[java.net.http.HttpResponse[Void]](
                    fx.Http.apply[java.net.http.HttpResponse[Void]](
                      fx.HEAD(uri)(timeout, headers).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[Void]], HttpScalaFXBackend.this.control, 
                        HttpScalaFXBackend.this.client
                      )
                    )
                  )[java.net.http.HttpResponse[fx.Receive[Byte]]](
                    {
                      {
                        def $anonfun(response: java.net.http.HttpResponse[Void]): java.net.http.HttpResponse[fx.Receive[Byte]] = 
                          {
                            {
                              final class $anon() extends Object(), java.net.http.HttpResponse[fx.Receive[Byte]] {
                                def statusCode(): Int = response.statusCode()
                                def request(): java.net.http.HttpRequest = response.request()
                                def previousResponse(): java.util.Optional[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
                                  java.util.Optional.ofNullable[java.net.http.HttpResponse[fx.Receive[Byte]]](null)
                                def headers(): java.net.http.HttpHeaders = response.headers()
                                def body(): fx.Receive[Byte] = fx.streamOf[Byte](wrapByteArray(Array.emptyByteArray).toList)
                                def sslSession(): java.util.Optional[javax.net.ssl.SSLSession] = response.sslSession()
                                def uri(): java.net.URI = response.uri()
                                def version(): java.net.http.HttpClient.Version = response.version()
                              }
                              new Object with java.net.http.HttpResponse {...}():java.net.http.HttpResponse[fx.Receive[Byte]]
                            }
                          }
                        closure($anonfun)
                      }
                    }
                  )
                this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                  fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](response)(HttpScalaFXBackend.this.control)
                )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
              }
           else ()
          if sttp.model.Method.OPTIONS.==(x21) then 
            return[matchResult6] 
              {
                this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                  fx.OPTIONS(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                    HttpScalaFXBackend.this.control
                  , HttpScalaFXBackend.this.client)
                )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
              }
           else ()
          if sttp.model.Method.PATCH.==(x21) then 
            return[matchResult6] 
              {
                this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                  fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](
                    sttp.fx.patch(uri)[fx.Receive[Byte]].apply[A](body, timeout, headers)(x$4, bm, HttpScalaFXBackend.this.control)
                  )(HttpScalaFXBackend.this.control)
                )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
              }
           else ()
          if sttp.model.Method.POST.==(x21) then 
            return[matchResult6] 
              {
                this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                  fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](
                    sttp.fx.post(uri)[fx.Receive[Byte]].apply[A](body, timeout, headers)(x$4, bm, HttpScalaFXBackend.this.control)
                  )(HttpScalaFXBackend.this.control)
                )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
              }
           else ()
          if sttp.model.Method.PUT.==(x21) then 
            return[matchResult6] 
              {
                this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                  fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](
                    sttp.fx.put(uri)[fx.Receive[Byte]].apply[A](body, timeout, headers)(x$4, bm, HttpScalaFXBackend.this.control)
                  )(HttpScalaFXBackend.this.control)
                )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
              }
           else ()
          if sttp.model.Method.TRACE.==(x21) then 
            return[matchResult6] 
              {
                this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                  fx.TRACE(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                    HttpScalaFXBackend.this.control
                  , HttpScalaFXBackend.this.client)
                )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
              }
           else ()
          case val m: sttp.client3.Identity[sttp.model.Method] = x21
          return[matchResult6] 
            {
              fx.Http.apply[sttp.client3.Response[T]](
                HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(new RuntimeException("Method: ".+(m).+(" is unsupported."))))[
                  sttp.client3.Response[T]
                ]
              )
            }
        }
    }
} of class class dotty.tools.dotc.ast.Trees$TypeDef # -1
exception while typing package sttp.fx {
  @SourceFile("sttp-scala-fx/src/main/scala/sttp/fx/HttpScalaFXBackend.scala") class HttpScalaFXBackend()(using client: fx.HttpClient, 
    control: fx.Control[Throwable | fx.HttpExecutionException]
  , config: fx.HttpClientConfig) extends Object(), sttp.client3.SttpBackend[fx.Http, sttp.fx.ReceiveStreams] {
    private given val client: fx.HttpClient
    private given val control: fx.Control[Throwable | fx.HttpExecutionException]
    private given val config: fx.HttpClientConfig
    private final lazy given def given_MonadError_Http: sttp.monad.MonadError[fx.Http] = this.responseMonad
    def send[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]](request: sttp.client3.Request[T, R]): fx.Http[sttp.client3.Response[T]] = 
      matchResult1[Function3]: 
        {
          case val x1: (request.body : (): sttp.client3.RequestBody) = request.body
          if x1.eq(sttp.client3.NoBody) then 
            {
              case val x: sttp.client3.NoBody.type = x1.$asInstanceOf[sttp.client3.NoBody.type]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.NoBody.type](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_NoBody_type
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.StringBody] then 
            {
              case val x: sttp.client3.StringBody = x1.$asInstanceOf[sttp.client3.StringBody]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.StringBody](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_StringBody
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.ByteArrayBody] then 
            {
              case val x: sttp.client3.ByteArrayBody = x1.$asInstanceOf[sttp.client3.ByteArrayBody]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.ByteArrayBody](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_ByteArrayBody
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.ByteBufferBody] then 
            {
              case val x: sttp.client3.ByteBufferBody = x1.$asInstanceOf[sttp.client3.ByteBufferBody]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.ByteBufferBody](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_ByteBufferBody
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.InputStreamBody] then 
            {
              case val x: sttp.client3.InputStreamBody = x1.$asInstanceOf[sttp.client3.InputStreamBody]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.InputStreamBody](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_InputStreamBody
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.FileBody] then 
            {
              case val x: sttp.client3.FileBody = x1.$asInstanceOf[sttp.client3.FileBody]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.FileBody](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_FileBody
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.StreamBody[BinaryStream$1, S$1]] then 
            {
              case val x6: sttp.client3.StreamBody[BinaryStream$1, S$1] = 
                sttp.client3.StreamBody.unapply[BinaryStream$1 @ BinaryStream$1, S$1 @ S$1](x1.$asInstanceOf[sttp.client3.StreamBody[BinaryStream$1, S$1]])
              case val x7: Any = x6._1
              case val x: sttp.client3.StreamBody[BinaryStream$1, S$1] = x1.$asInstanceOf[sttp.client3.StreamBody[BinaryStream$1, S$1]]
              return[matchResult1] 
                {
                  val body: sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams] = 
                    {
                      val $proxy1: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                        fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                      (try 
                        {
                          {
                            def $anonfun(using evidence$1: fx.Throws[Exception]): sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams] = 
                              x.asInstanceOf[sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]]
                            closure($anonfun)
                          }.$asInstanceOf[
                            ((fx.Throws[Exception]) ?=> sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]) & (($proxy1.Throws[Exception]) 
                              ?=>
                             sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams])
                          ].apply(fx.Throws.unsafeExceptions[Exception])
                        }
                       catch 
                        {
                          case r @ _:Exception => 
                            {
                              HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[
                                sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]
                              ]
                            }
                        }
                      ):sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]
                    }
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), 
                    sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]
                  ](request, body)(sttp.fx.given_ToHttpBodyMapper_StreamBody, fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.MultipartBody[R$1]] then 
            {
              case val x3: sttp.client3.MultipartBody[R$1] = 
                sttp.client3.MultipartBody.unapply[R$1 @ R$1](x1.$asInstanceOf[sttp.client3.MultipartBody[R$1]])
              case val x4: scala.collection.immutable.Seq[sttp.model.Part[sttp.client3.RequestBody[R$1]]] = x3._1
              case val x: sttp.client3.MultipartBody[R$1] = x1.$asInstanceOf[sttp.client3.MultipartBody[R$1]]
              return[matchResult1] 
                {
                  val body: sttp.client3.MultipartBody[R] = 
                    {
                      val $proxy2: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                        fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                      (try 
                        {
                          {
                            def $anonfun(using evidence$2: fx.Throws[Exception]): sttp.client3.MultipartBody[R] = 
                              x.asInstanceOf[sttp.client3.MultipartBody[R]]
                            closure($anonfun)
                          }.$asInstanceOf[
                            ((fx.Throws[Exception]) ?=> sttp.client3.MultipartBody[R]) & (($proxy2.Throws[Exception]) ?=> sttp.client3.MultipartBody[R])
                          ].apply(fx.Throws.unsafeExceptions[Exception])
                        }
                       catch 
                        {
                          case r @ _:Exception => 
                            {
                              HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[sttp.client3.MultipartBody[R]]
                            }
                        }
                      ):sttp.client3.MultipartBody[R]
                    }
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.MultipartBody[R]](request, body)(
                    sttp.fx.multipartBodyMapper[R]
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          throw new MatchError(x1)
        }
    def close(): fx.Http[Unit] = ???
    def responseMonad: sttp.monad.MonadError[fx.Http] = 
      {
        final class $anon() extends Object(), sttp.monad.MonadError[fx.Http] {
          def ensure[T](f: fx.Http[T], e: () => fx.Http[Unit]): fx.Http[T] = 
            {
              val x: Throwable | T = 
                fx.run[Exception, T](
                  {
                    def $anonfun(using evidence$3: fx.Control[Exception]): T = fx.Http.value[T](f)(evidence$3)
                    closure($anonfun)
                  }
                )
              matchResult2[Function3]: 
                {
                  case val x14: (x : Object) = x
                  if x14.$isInstanceOf[Throwable] then 
                    {
                      case val ex: Throwable = x14.$asInstanceOf[Throwable]
                      return[matchResult2] 
                        {
                          val ignored: fx.Http[Unit] = 
                            fx.run[Nothing, fx.Http[Unit]](
                              {
                                def $anonfun(using evidence$4: fx.Control[Nothing]): fx.Http[Unit] = e.apply()
                                closure($anonfun)
                              }
                            )
                          f
                        }
                    }
                   else ()
                  case val a: Throwable | T = x14
                  return[matchResult2] 
                    {
                      this.unit[T](a.asInstanceOf[T])
                    }
                }
            }
          def error[T](t: Throwable): fx.Http[T] = sttp.fx.HttpScalaFXBackend.this.control.shift(t)[Nothing]
          def flatMap[T, T2](fa: fx.Http[T])(f: T => fx.Http[T2]): fx.Http[T2] = fx.Http.flatMap[T](fa)[T2](f)
          protected def handleWrappedError[T](rt: fx.Http[T])(h: PartialFunction[Throwable, fx.Http[T]]): fx.Http[T] = 
            {
              val x: Throwable | T = 
                fx.run[Exception, T](
                  {
                    def $anonfun(using evidence$5: fx.Control[Exception]): T = fx.Http.value[T](rt)(evidence$5)
                    closure($anonfun)
                  }
                )
              matchResult3[Function3]: 
                {
                  case val x16: (x : Object) = x
                  if x16.$isInstanceOf[Throwable] then 
                    {
                      case val x18: Throwable = x16.$asInstanceOf[Throwable]
                      {
                        case val ex: Throwable = x18
                        if h.isDefinedAt(ex) then 
                          return[matchResult3] 
                            {
                              h.apply(ex)
                            }
                         else ()
                      }
                      case val ex: Throwable = x18
                      return[matchResult3] 
                        {
                          fx.raise[fx.HttpExecutionException](fx.HttpExecutionException.apply(new RuntimeException(ex)))[Nothing](
                            sttp.fx.HttpScalaFXBackend.this.control
                          )
                        }
                    }
                   else ()
                  case val a: Throwable | T = x16
                  return[matchResult3] 
                    {
                      this.unit[T](a.asInstanceOf[T])
                    }
                }
            }
          def map[T, T2](fa: fx.Http[T])(f: T => T2): fx.Http[T2] = fx.Http.map[T](fa)[T2](f)
          def unit[T](t: T): fx.Http[T] = fx.Http.apply[T](t)
          private val $outer: sttp.fx.HttpScalaFXBackend
          final def sttp$fx$HttpScalaFXBackend$_$$anon$$$outer: sttp.fx.HttpScalaFXBackend = $outer
        }
        new Object with sttp.monad.MonadError {...}():sttp.monad.MonadError[fx.Http]
      }
    private def requestAsBody[R, A <: sttp.client3.RequestBody[R]](r: A)(using x$2: sttp.fx.ToHttpBodyMapper[A]): fx.Http[fx.HttpBodyMapper[A]] = 
      {
        val $proxy3: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
          fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
        (try 
          {
            {
              def $anonfun(using evidence$6: fx.Throws[Exception]): fx.Http[fx.HttpBodyMapper[A]] = 
                fx.Http.apply[fx.HttpBodyMapper[A]](x$2.toHttpBodyMapper(r)())
              closure($anonfun)
            }.$asInstanceOf[((fx.Throws[Exception]) ?=> fx.Http[fx.HttpBodyMapper[A]]) & (($proxy3.Throws[Exception]) ?=> fx.Http[fx.HttpBodyMapper[A]])].
              apply
            (fx.Throws.unsafeExceptions[Exception])
          }
         catch 
          {
            case r @ _:Exception => 
              {
                def $anonfun(e: Exception): fx.Http[fx.HttpBodyMapper[A]] = 
                  HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(e))[Nothing]
                closure($anonfun)
              }.$asInstanceOf[Exception => fx.Http[fx.HttpBodyMapper[A]]].apply(r)
          }
        ):fx.Http[fx.HttpBodyMapper[A]]
      }
    private def getRequestHeaders[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http], A](request: sttp.client3.Request[T, R], body: A)(
      using 
    bodyMapper: fx.HttpBodyMapper[A]): fx.Http[scala.collection.immutable.::[fx.HttpHeader]] = 
      {
        val Content-Type: fx.HttpHeader = 
          fx.HttpHeader.apply(
            {
              val ev$1: ArrowAssoc[String] = ArrowAssoc[String]("Content-Type")
              scala.Predef.ArrowAssoc.->$extension[scala.collection.immutable.::[String], String](ev$1)(
                ::.apply[String](fx.MediaType.value(bodyMapper.mediaType), List.empty[String])
              )
            }
          )
        fx.Http.apply[scala.collection.immutable.::[fx.HttpHeader]](
          ::.apply[(String, scala.collection.immutable.::[String])](`Content-Type`, 
            request.headers.filterNot(
              {
                def $anonfun(_$1: sttp.model.Header): Boolean = _$1.name.==("Content-Length").unary_!
                closure($anonfun)
              }
            ).groupBy[String](
              {
                def $anonfun(_$2: sttp.model.Header): String = _$2.name
                closure($anonfun)
              }
            ).map[String, scala.collection.immutable.::[String]](
              {
                {
                  def $anonfun(p: (String, scala.collection.immutable.Seq[sttp.model.Header])): fx.HttpHeader = 
                    {
                      val values: scala.collection.immutable.List[String] = 
                        p._2.map[String](
                          {
                            def $anonfun(_$3: sttp.model.Header): String = _$3.value
                            closure($anonfun)
                          }
                        ).toList
                      fx.HttpHeader.apply(p._1, 
                        values.headOption.getOrElse[String](
                          <special-ops>.<cbn-arg>[("" : String)](
                            {
                              def $anonfun(): ("" : String) = ""
                              closure($anonfun)
                            }
                          )
                        )
                      , values.tail)
                    }
                  closure($anonfun)
                }
              }
            ).toList
          )
        )
      }
    private def getResponseHeaders[T](response: java.net.http.HttpResponse[T]): fx.Http[scala.collection.immutable.Seq[sttp.model.Header]] = 
      {
        fx.Http.apply[scala.collection.immutable.Seq[sttp.model.Header]](
          jdk.CollectionConverters.MapHasAsScala[String, java.util.List[String]](response.headers().map()).asScala.toList.foldLeft[
            scala.collection.immutable.Seq[sttp.model.Header]
          ](Seq.empty[sttp.model.Header])(
            {
              {
                def $anonfun(acc: scala.collection.immutable.Seq[sttp.model.Header], jnhHeader: (String, java.util.List[String])): 
                  scala.collection.immutable.Seq[sttp.model.Header]
                 = 
                  {
                    acc.++[sttp.model.Header](
                      jdk.CollectionConverters.ListHasAsScala[String](jnhHeader._2).asScala.map[sttp.model.Header](
                        {
                          def $anonfun(_$4: String): sttp.model.Header = sttp.model.Header.apply(jnhHeader._1, _$4)
                          closure($anonfun)
                        }
                      )
                    )
                  }
                closure($anonfun)
              }
            }
          )
        )
      }
    private def bodyAsResponseAs[T](using x$1: sttp.monad.MonadError[fx.Http]): 
      sttp.client3.internal.BodyFromResponseAs[fx.Http, java.net.http.HttpResponse[fx.Receive[Byte]], Nothing, fx.Receive[Byte]]
     = 
      {
        final class $anon() extends 
          sttp.client3.internal.BodyFromResponseAs[fx.Http, java.net.http.HttpResponse[fx.Receive[Byte]], Nothing, fx.Receive[Byte]]
        ()(x$1) {
          protected override def withReplayableBody(response: java.net.http.HttpResponse[fx.Receive[Byte]], 
            replayableBody: scala.util.Either[Array[Byte], sttp.client3.internal.SttpFile]
          ): fx.Http[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
            fx.Http.apply[java.net.http.HttpResponse[fx.Receive[Byte]]](
              {
                final class $anon() extends Object(), java.net.http.HttpResponse[fx.Receive[Byte]] {
                  def statusCode(): Int = response.statusCode()
                  def request(): java.net.http.HttpRequest = response.request()
                  def previousResponse(): java.util.Optional[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
                    java.util.Optional.of[java.net.http.HttpResponse[fx.Receive[Byte]]](response)
                  def headers(): java.net.http.HttpHeaders = response.headers()
                  def body(): fx.Receive[Byte] = 
                    replayableBody.fold[fx.Receive[Byte]](
                      {
                        def $anonfun(bytes: Array[Byte]): fx.Receive[Byte] = fx.streamOf[Byte](wrapByteArray(bytes).toList)
                        closure($anonfun)
                      }
                    , 
                      {
                        def $anonfun(file: sttp.client3.internal.SttpFile): fx.Receive[Byte] = 
                          fx.streamOf[Byte](wrapByteArray(file.readAsByteArray).toList)
                        closure($anonfun)
                      }
                    )
                  def sslSession(): java.util.Optional[javax.net.ssl.SSLSession] = response.sslSession()
                  def uri(): java.net.URI = response.uri()
                  def version(): java.net.http.HttpClient.Version = response.version()
                }
                new Object with java.net.http.HttpResponse {...}():java.net.http.HttpResponse[fx.Receive[Byte]]
              }
            )
          protected override def regularIgnore(response: java.net.http.HttpResponse[fx.Receive[Byte]]): fx.Http[Unit] = 
            fx.Http.map[fx.Receive[Byte]](
              fx.Http.apply[fx.Receive[Byte]](
                {
                  val $proxy4: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                    fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                  {
                    def liftedTree1(): fx.Receive[Byte] = 
                      (try 
                        {
                          {
                            def $anonfun(using evidence$7: fx.Throws[Exception]): fx.Receive[Byte] = response.body()
                            closure($anonfun)
                          }.$asInstanceOf[((fx.Throws[Exception]) ?=> fx.Receive[Byte]) & (($proxy4.Throws[Exception]) ?=> fx.Receive[Byte])].apply(
                            fx.Throws.unsafeExceptions[Exception]
                          )
                        }
                       catch 
                        {
                          case r @ _:Exception => 
                            {
                              sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[fx.Receive[Byte]]
                            }
                        }
                      )
                    liftedTree1()
                  }:fx.Receive[Byte]
                }
              )
            )[Unit](
              {
                def $anonfun(x$1: fx.Receive[Byte]): Unit = 
                  matchResult4[Unit]: 
                    {
                      case val x19: (x$1 : fx.Receive) @unchecked = x$1:(x$1 : fx.Receive) @unchecked
                      return[matchResult4] 
                        {
                          fx.Http.apply[Unit](())
                          ()
                        }
                    }
                closure($anonfun)
              }
            )
          protected override def regularAsByteArray(response: java.net.http.HttpResponse[fx.Receive[Byte]]): fx.Http[Array[Byte]] = 
            fx.Http.apply[Array[Byte]](
              {
                val $proxy5: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                  fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                {
                  def liftedTree2(): Array[Byte] = 
                    (try 
                      {
                        {
                          def $anonfun(using evidence$8: fx.Throws[Exception]): Array[Byte] = 
                            fx.toList[Byte](response.body()).toArray[Byte](scala.reflect.ClassTag.apply[Byte](classOf[Byte]))
                          closure($anonfun)
                        }.$asInstanceOf[((fx.Throws[Exception]) ?=> Array[Byte]) & (($proxy5.Throws[Exception]) ?=> Array[Byte])].apply(
                          fx.Throws.unsafeExceptions[Exception]
                        )
                      }
                     catch 
                      {
                        case r @ _:Exception => 
                          {
                            sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[Array[Byte]]
                          }
                      }
                    )
                  liftedTree2()
                }:Array[Byte]
              }
            )
          protected override def regularAsFile(response: java.net.http.HttpResponse[fx.Receive[Byte]], file: sttp.client3.internal.SttpFile): 
            fx.Http[sttp.client3.internal.SttpFile]
           = 
            fx.Http.map[java.nio.file.Path](
              fx.Http.apply[java.nio.file.Path](
                {
                  val $proxy6: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                    fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                  {
                    def liftedTree3(): java.nio.file.Path = 
                      (try 
                        {
                          {
                            def $anonfun(using evidence$9: fx.Throws[Exception]): java.nio.file.Path = 
                              java.nio.file.Files.write(file.toPath, 
                                fx.toList[Byte](response.body()).toArray[Byte](scala.reflect.ClassTag.apply[Byte](classOf[Byte]))
                              , [ : java.nio.file.OpenOption])
                            closure($anonfun)
                          }.$asInstanceOf[((fx.Throws[Exception]) ?=> java.nio.file.Path) & (($proxy6.Throws[Exception]) ?=> java.nio.file.Path)].apply(
                            fx.Throws.unsafeExceptions[Exception]
                          )
                        }
                       catch 
                        {
                          case r @ _:Exception => 
                            {
                              sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[java.nio.file.Path]
                            }
                        }
                      )
                    liftedTree3()
                  }:java.nio.file.Path
                }
              )
            )[sttp.client3.internal.SttpFile](
              {
                def $anonfun(x$1: java.nio.file.Path): sttp.client3.internal.SttpFile = 
                  matchResult5[sttp.client3.internal.SttpFile]: 
                    {
                      case val x20: (x$1 : java.nio.file.Path) @unchecked = x$1:(x$1 : java.nio.file.Path) @unchecked
                      return[matchResult5] file
                    }
                closure($anonfun)
              }
            )
          protected override def regularAsStream(response: java.net.http.HttpResponse[fx.Receive[Byte]]): fx.Http[(fx.Receive[Byte], () => fx.Http[Unit])]
             = 
          fx.Http.apply[(fx.Receive[Byte], () => fx.Http[Unit])](
            {
              val $proxy7: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
              {
                def liftedTree4(): (fx.Receive[Byte], () => fx.Http[Unit]) = 
                  (try 
                    {
                      {
                        def $anonfun(using evidence$10: fx.Throws[Exception]): (fx.Receive[Byte], () => fx.Http[Unit]) = 
                          Tuple2.apply[fx.Receive[Byte], () => fx.Http[Unit]](response.body(), 
                            {
                              def $anonfun(): fx.Http[Unit] = fx.Http.apply[Unit](())
                              closure($anonfun)
                            }
                          )
                        closure($anonfun)
                      }.$asInstanceOf[
                        ((fx.Throws[Exception]) ?=> (fx.Receive[Byte], () => fx.Http[Unit])) & (($proxy7.Throws[Exception]) ?=> (fx.Receive[Byte], () => 
                          fx.Http[Unit]
                        ))
                      ].apply(fx.Throws.unsafeExceptions[Exception])
                    }
                   catch 
                    {
                      case r @ _:Exception => 
                        {
                          sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[(fx.Receive[Byte], () => fx.Http[Unit])]
                        }
                    }
                  )
                liftedTree4()
              }:(fx.Receive[Byte], () => fx.Http[Unit])
            }
          )
          protected override def handleWS[T](responseAs: sttp.client3.WebSocketResponseAs[T, ?], meta: sttp.model.ResponseMetadata, ws: Nothing): 
            fx.Http[T]
           = 
            fx.Http.apply[T](
              sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(new RuntimeException("Websockets are unsupported")))[T]
            )
          protected override def cleanupWhenNotAWebSocket(response: java.net.http.HttpResponse[fx.Receive[Byte]], e: sttp.client3.ws.NotAWebSocketException
            )
          : fx.Http[Unit] = fx.Http.apply[Unit](())
          protected override def cleanupWhenGotWebSocket(response: Nothing, e: sttp.client3.ws.GotAWebSocketException): fx.Http[Unit] = 
            fx.Http.apply[Unit](
              sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(new RuntimeException("Websockets are unsupported")))[Unit]
            )
          private val $outer: sttp.fx.HttpScalaFXBackend
          final def sttp$fx$HttpScalaFXBackend$_$$anon$$$outer: sttp.fx.HttpScalaFXBackend = $outer
        }
        new sttp.client3.internal.BodyFromResponseAs {...}():
          sttp.client3.internal.BodyFromResponseAs[fx.Http, java.net.http.HttpResponse[fx.Receive[Byte]], Nothing, fx.Receive[Byte]]
      }
    private def toResponse[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]](request: sttp.client3.Request[T, R], 
      response: java.net.http.HttpResponse[fx.Receive[Byte]]
    )(using x$3: sttp.fx.StatusCodeToStatusCode[sttp.model.StatusCode, fx.StatusCode], x$4: sttp.fx.StatusCodeToStatusCode[Int, sttp.model.StatusCode]): 
      fx.Http[sttp.client3.Response[T]]
     = 
      {
        val status: sttp.model.StatusCode = 
          fx.Http.value[sttp.model.StatusCode](x$4.toStatusCode(response.statusCode())()(HttpScalaFXBackend.this.control))(HttpScalaFXBackend.this.control)
        val statusText: String = 
          fx.StatusCode.statusText(
            fx.Http.value[fx.StatusCode](x$3.toStatusCode(status)()(HttpScalaFXBackend.this.control))(HttpScalaFXBackend.this.control)
          )
        val headers: scala.collection.immutable.Seq[sttp.model.Header] = 
          fx.Http.value[scala.collection.immutable.Seq[sttp.model.Header]](this.getResponseHeaders[fx.Receive[Byte]](response))(
            HttpScalaFXBackend.this.control
          )
        val responseMetadata: sttp.model.ResponseMetadata = sttp.model.ResponseMetadata.apply(status, statusText, headers)
        val body: T = 
          fx.Http.value[T](
            this.bodyAsResponseAs[T](this.given_MonadError_Http).apply[T](request.response, responseMetadata, 
              Left.apply[java.net.http.HttpResponse[fx.Receive[Byte]], Nothing](response)
            )
          )(HttpScalaFXBackend.this.control)
        fx.Http.apply[sttp.client3.Response[T]](sttp.client3.Response.apply[T](body, status, statusText, headers))
      }
    private def getUri[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]](request: sttp.client3.Request[T, R]): fx.Http[java.net.URI] = 
      {
        fx.Http.apply[java.net.URI](
          {
            val $proxy8: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
              fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
            {
              def liftedTree5(): java.net.URI = 
                (try 
                  {
                    {
                      def $anonfun(using evidence$11: fx.Throws[Exception]): java.net.URI = request.uri.toJavaUri
                      closure($anonfun)
                    }.$asInstanceOf[((fx.Throws[Exception]) ?=> java.net.URI) & (($proxy8.Throws[Exception]) ?=> java.net.URI)].apply(
                      fx.Throws.unsafeExceptions[Exception]
                    )
                  }
                 catch 
                  {
                    case r @ _:Exception => 
                      {
                        def $anonfun(e: Exception): java.net.URI = HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(e))[Nothing]
                        closure($anonfun)
                      }.$asInstanceOf[Exception => java.net.URI].apply(r)
                  }
                )
              liftedTree5()
            }:java.net.URI
          }
        )
      }
    private def makeRequest[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http], A](request: sttp.client3.Request[T, R], body: A)(using 
      x$3: sttp.fx.ToHttpBodyMapper[A]
    , x$4: fx.HttpResponseMapper[fx.Receive[Byte]]): fx.Http[sttp.client3.Response[T]] = 
      {
        final lazy given def bm: fx.HttpBodyMapper[A] = x$3.toHttpBodyMapper(body)()
        val headers: scala.collection.immutable.::[fx.HttpHeader] = 
          fx.Http.value[scala.collection.immutable.::[fx.HttpHeader]](
            this.getRequestHeaders[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), A](request, body)(bm)
          )(HttpScalaFXBackend.this.control)
        val uri: java.net.URI = 
          fx.Http.value[java.net.URI](this.getUri[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request))(
            HttpScalaFXBackend.this.control
          )
        val timeout: scala.concurrent.duration.Duration = request.options.readTimeout
        matchResult6[Function3]: 
          {
            case val x21: (request.method : (): Object) = request.method
            if sttp.model.Method.DELETE.==(x21) then 
              return[matchResult6] 
                {
                  this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                    fx.DELETE(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                      HttpScalaFXBackend.this.control
                    , HttpScalaFXBackend.this.client)
                  )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                }
             else ()
            if sttp.model.Method.GET.==(x21) then 
              return[matchResult6] 
                {
                  this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                    fx.GET(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                      HttpScalaFXBackend.this.control
                    , HttpScalaFXBackend.this.client)
                  )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                }
             else ()
            if sttp.model.Method.HEAD.==(x21) then 
              return[matchResult6] 
                {
                  val response: fx.Http[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
                    fx.Http.map[java.net.http.HttpResponse[Void]](
                      fx.Http.apply[java.net.http.HttpResponse[Void]](
                        fx.HEAD(uri)(timeout, headers).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[Void]], HttpScalaFXBackend.this.control, 
                          HttpScalaFXBackend.this.client
                        )
                      )
                    )[java.net.http.HttpResponse[fx.Receive[Byte]]](
                      {
                        {
                          def $anonfun(response: java.net.http.HttpResponse[Void]): java.net.http.HttpResponse[fx.Receive[Byte]] = 
                            {
                              {
                                final class $anon() extends Object(), java.net.http.HttpResponse[fx.Receive[Byte]] {
                                  def statusCode(): Int = response.statusCode()
                                  def request(): java.net.http.HttpRequest = response.request()
                                  def previousResponse(): java.util.Optional[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
                                    java.util.Optional.ofNullable[java.net.http.HttpResponse[fx.Receive[Byte]]](null)
                                  def headers(): java.net.http.HttpHeaders = response.headers()
                                  def body(): fx.Receive[Byte] = fx.streamOf[Byte](wrapByteArray(Array.emptyByteArray).toList)
                                  def sslSession(): java.util.Optional[javax.net.ssl.SSLSession] = response.sslSession()
                                  def uri(): java.net.URI = response.uri()
                                  def version(): java.net.http.HttpClient.Version = response.version()
                                }
                                new Object with java.net.http.HttpResponse {...}():java.net.http.HttpResponse[fx.Receive[Byte]]
                              }
                            }
                          closure($anonfun)
                        }
                      }
                    )
                  this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                    fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](response)(HttpScalaFXBackend.this.control)
                  )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                }
             else ()
            if sttp.model.Method.OPTIONS.==(x21) then 
              return[matchResult6] 
                {
                  this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                    fx.OPTIONS(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                      HttpScalaFXBackend.this.control
                    , HttpScalaFXBackend.this.client)
                  )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                }
             else ()
            if sttp.model.Method.PATCH.==(x21) then 
              return[matchResult6] 
                {
                  this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                    fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](
                      sttp.fx.patch(uri)[fx.Receive[Byte]].apply[A](body, timeout, headers)(x$4, bm, HttpScalaFXBackend.this.control)
                    )(HttpScalaFXBackend.this.control)
                  )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                }
             else ()
            if sttp.model.Method.POST.==(x21) then 
              return[matchResult6] 
                {
                  this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                    fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](
                      sttp.fx.post(uri)[fx.Receive[Byte]].apply[A](body, timeout, headers)(x$4, bm, HttpScalaFXBackend.this.control)
                    )(HttpScalaFXBackend.this.control)
                  )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                }
             else ()
            if sttp.model.Method.PUT.==(x21) then 
              return[matchResult6] 
                {
                  this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                    fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](
                      sttp.fx.put(uri)[fx.Receive[Byte]].apply[A](body, timeout, headers)(x$4, bm, HttpScalaFXBackend.this.control)
                    )(HttpScalaFXBackend.this.control)
                  )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                }
             else ()
            if sttp.model.Method.TRACE.==(x21) then 
              return[matchResult6] 
                {
                  this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                    fx.TRACE(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                      HttpScalaFXBackend.this.control
                    , HttpScalaFXBackend.this.client)
                  )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                }
             else ()
            case val m: sttp.client3.Identity[sttp.model.Method] = x21
            return[matchResult6] 
              {
                fx.Http.apply[sttp.client3.Response[T]](
                  HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(new RuntimeException("Method: ".+(m).+(" is unsupported."))))[
                    sttp.client3.Response[T]
                  ]
                )
              }
          }
      }
  }
} of class class dotty.tools.dotc.ast.Trees$PackageDef # -1
exception while typing package sttp {
  package sttp.fx {
    @SourceFile("sttp-scala-fx/src/main/scala/sttp/fx/HttpScalaFXBackend.scala") class HttpScalaFXBackend()(using client: fx.HttpClient, 
      control: fx.Control[Throwable | fx.HttpExecutionException]
    , config: fx.HttpClientConfig) extends Object(), sttp.client3.SttpBackend[fx.Http, sttp.fx.ReceiveStreams] {
      private given val client: fx.HttpClient
      private given val control: fx.Control[Throwable | fx.HttpExecutionException]
      private given val config: fx.HttpClientConfig
      private final lazy given def given_MonadError_Http: sttp.monad.MonadError[fx.Http] = this.responseMonad
      def send[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]](request: sttp.client3.Request[T, R]): fx.Http[sttp.client3.Response[T]]
         = 
      matchResult1[Function3]: 
        {
          case val x1: (request.body : (): sttp.client3.RequestBody) = request.body
          if x1.eq(sttp.client3.NoBody) then 
            {
              case val x: sttp.client3.NoBody.type = x1.$asInstanceOf[sttp.client3.NoBody.type]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.NoBody.type](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_NoBody_type
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.StringBody] then 
            {
              case val x: sttp.client3.StringBody = x1.$asInstanceOf[sttp.client3.StringBody]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.StringBody](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_StringBody
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.ByteArrayBody] then 
            {
              case val x: sttp.client3.ByteArrayBody = x1.$asInstanceOf[sttp.client3.ByteArrayBody]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.ByteArrayBody](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_ByteArrayBody
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.ByteBufferBody] then 
            {
              case val x: sttp.client3.ByteBufferBody = x1.$asInstanceOf[sttp.client3.ByteBufferBody]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.ByteBufferBody](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_ByteBufferBody
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.InputStreamBody] then 
            {
              case val x: sttp.client3.InputStreamBody = x1.$asInstanceOf[sttp.client3.InputStreamBody]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.InputStreamBody](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_InputStreamBody
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.FileBody] then 
            {
              case val x: sttp.client3.FileBody = x1.$asInstanceOf[sttp.client3.FileBody]
              return[matchResult1] 
                {
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.FileBody](request, x)(
                    sttp.fx.given_ToHttpBodyMapper_FileBody
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.StreamBody[BinaryStream$1, S$1]] then 
            {
              case val x6: sttp.client3.StreamBody[BinaryStream$1, S$1] = 
                sttp.client3.StreamBody.unapply[BinaryStream$1 @ BinaryStream$1, S$1 @ S$1](x1.$asInstanceOf[sttp.client3.StreamBody[BinaryStream$1, S$1]])
              case val x7: Any = x6._1
              case val x: sttp.client3.StreamBody[BinaryStream$1, S$1] = x1.$asInstanceOf[sttp.client3.StreamBody[BinaryStream$1, S$1]]
              return[matchResult1] 
                {
                  val body: sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams] = 
                    {
                      val $proxy1: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                        fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                      (try 
                        {
                          {
                            def $anonfun(using evidence$1: fx.Throws[Exception]): sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams] = 
                              x.asInstanceOf[sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]]
                            closure($anonfun)
                          }.$asInstanceOf[
                            ((fx.Throws[Exception]) ?=> sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]) & (($proxy1.Throws[Exception]) 
                              ?=>
                             sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams])
                          ].apply(fx.Throws.unsafeExceptions[Exception])
                        }
                       catch 
                        {
                          case r @ _:Exception => 
                            {
                              HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[
                                sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]
                              ]
                            }
                        }
                      ):sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]
                    }
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), 
                    sttp.client3.StreamBody[fx.Receive[Byte], sttp.fx.ReceiveStreams]
                  ](request, body)(sttp.fx.given_ToHttpBodyMapper_StreamBody, fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          if x1.$isInstanceOf[sttp.client3.MultipartBody[R$1]] then 
            {
              case val x3: sttp.client3.MultipartBody[R$1] = 
                sttp.client3.MultipartBody.unapply[R$1 @ R$1](x1.$asInstanceOf[sttp.client3.MultipartBody[R$1]])
              case val x4: scala.collection.immutable.Seq[sttp.model.Part[sttp.client3.RequestBody[R$1]]] = x3._1
              case val x: sttp.client3.MultipartBody[R$1] = x1.$asInstanceOf[sttp.client3.MultipartBody[R$1]]
              return[matchResult1] 
                {
                  val body: sttp.client3.MultipartBody[R] = 
                    {
                      val $proxy2: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                        fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                      (try 
                        {
                          {
                            def $anonfun(using evidence$2: fx.Throws[Exception]): sttp.client3.MultipartBody[R] = 
                              x.asInstanceOf[sttp.client3.MultipartBody[R]]
                            closure($anonfun)
                          }.$asInstanceOf[
                            ((fx.Throws[Exception]) ?=> sttp.client3.MultipartBody[R]) & (($proxy2.Throws[Exception]) ?=> sttp.client3.MultipartBody[R])
                          ].apply(fx.Throws.unsafeExceptions[Exception])
                        }
                       catch 
                        {
                          case r @ _:Exception => 
                            {
                              HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[sttp.client3.MultipartBody[R]]
                            }
                        }
                      ):sttp.client3.MultipartBody[R]
                    }
                  this.makeRequest[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), sttp.client3.MultipartBody[R]](request, body)(
                    sttp.fx.multipartBodyMapper[R]
                  , fx.HttpResponseMapper.HttpResponseReceiveMapper)
                }
            }
           else ()
          throw new MatchError(x1)
        }
      def close(): fx.Http[Unit] = ???
      def responseMonad: sttp.monad.MonadError[fx.Http] = 
        {
          final class $anon() extends Object(), sttp.monad.MonadError[fx.Http] {
            def ensure[T](f: fx.Http[T], e: () => fx.Http[Unit]): fx.Http[T] = 
              {
                val x: Throwable | T = 
                  fx.run[Exception, T](
                    {
                      def $anonfun(using evidence$3: fx.Control[Exception]): T = fx.Http.value[T](f)(evidence$3)
                      closure($anonfun)
                    }
                  )
                matchResult2[Function3]: 
                  {
                    case val x14: (x : Object) = x
                    if x14.$isInstanceOf[Throwable] then 
                      {
                        case val ex: Throwable = x14.$asInstanceOf[Throwable]
                        return[matchResult2] 
                          {
                            val ignored: fx.Http[Unit] = 
                              fx.run[Nothing, fx.Http[Unit]](
                                {
                                  def $anonfun(using evidence$4: fx.Control[Nothing]): fx.Http[Unit] = e.apply()
                                  closure($anonfun)
                                }
                              )
                            f
                          }
                      }
                     else ()
                    case val a: Throwable | T = x14
                    return[matchResult2] 
                      {
                        this.unit[T](a.asInstanceOf[T])
                      }
                  }
              }
            def error[T](t: Throwable): fx.Http[T] = sttp.fx.HttpScalaFXBackend.this.control.shift(t)[Nothing]
            def flatMap[T, T2](fa: fx.Http[T])(f: T => fx.Http[T2]): fx.Http[T2] = fx.Http.flatMap[T](fa)[T2](f)
            protected def handleWrappedError[T](rt: fx.Http[T])(h: PartialFunction[Throwable, fx.Http[T]]): fx.Http[T] = 
              {
                val x: Throwable | T = 
                  fx.run[Exception, T](
                    {
                      def $anonfun(using evidence$5: fx.Control[Exception]): T = fx.Http.value[T](rt)(evidence$5)
                      closure($anonfun)
                    }
                  )
                matchResult3[Function3]: 
                  {
                    case val x16: (x : Object) = x
                    if x16.$isInstanceOf[Throwable] then 
                      {
                        case val x18: Throwable = x16.$asInstanceOf[Throwable]
                        {
                          case val ex: Throwable = x18
                          if h.isDefinedAt(ex) then 
                            return[matchResult3] 
                              {
                                h.apply(ex)
                              }
                           else ()
                        }
                        case val ex: Throwable = x18
                        return[matchResult3] 
                          {
                            fx.raise[fx.HttpExecutionException](fx.HttpExecutionException.apply(new RuntimeException(ex)))[Nothing](
                              sttp.fx.HttpScalaFXBackend.this.control
                            )
                          }
                      }
                     else ()
                    case val a: Throwable | T = x16
                    return[matchResult3] 
                      {
                        this.unit[T](a.asInstanceOf[T])
                      }
                  }
              }
            def map[T, T2](fa: fx.Http[T])(f: T => T2): fx.Http[T2] = fx.Http.map[T](fa)[T2](f)
            def unit[T](t: T): fx.Http[T] = fx.Http.apply[T](t)
            private val $outer: sttp.fx.HttpScalaFXBackend
            final def sttp$fx$HttpScalaFXBackend$_$$anon$$$outer: sttp.fx.HttpScalaFXBackend = $outer
          }
          new Object with sttp.monad.MonadError {...}():sttp.monad.MonadError[fx.Http]
        }
      private def requestAsBody[R, A <: sttp.client3.RequestBody[R]](r: A)(using x$2: sttp.fx.ToHttpBodyMapper[A]): fx.Http[fx.HttpBodyMapper[A]] = 
        {
          val $proxy3: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
            fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
          (try 
            {
              {
                def $anonfun(using evidence$6: fx.Throws[Exception]): fx.Http[fx.HttpBodyMapper[A]] = 
                  fx.Http.apply[fx.HttpBodyMapper[A]](x$2.toHttpBodyMapper(r)())
                closure($anonfun)
              }.$asInstanceOf[((fx.Throws[Exception]) ?=> fx.Http[fx.HttpBodyMapper[A]]) & (($proxy3.Throws[Exception]) ?=> fx.Http[fx.HttpBodyMapper[A]])]
                .
              apply(fx.Throws.unsafeExceptions[Exception])
            }
           catch 
            {
              case r @ _:Exception => 
                {
                  def $anonfun(e: Exception): fx.Http[fx.HttpBodyMapper[A]] = 
                    HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(e))[Nothing]
                  closure($anonfun)
                }.$asInstanceOf[Exception => fx.Http[fx.HttpBodyMapper[A]]].apply(r)
            }
          ):fx.Http[fx.HttpBodyMapper[A]]
        }
      private def getRequestHeaders[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http], A](request: sttp.client3.Request[T, R], body: A)(
        using 
      bodyMapper: fx.HttpBodyMapper[A]): fx.Http[scala.collection.immutable.::[fx.HttpHeader]] = 
        {
          val Content-Type: fx.HttpHeader = 
            fx.HttpHeader.apply(
              {
                val ev$1: ArrowAssoc[String] = ArrowAssoc[String]("Content-Type")
                scala.Predef.ArrowAssoc.->$extension[scala.collection.immutable.::[String], String](ev$1)(
                  ::.apply[String](fx.MediaType.value(bodyMapper.mediaType), List.empty[String])
                )
              }
            )
          fx.Http.apply[scala.collection.immutable.::[fx.HttpHeader]](
            ::.apply[(String, scala.collection.immutable.::[String])](`Content-Type`, 
              request.headers.filterNot(
                {
                  def $anonfun(_$1: sttp.model.Header): Boolean = _$1.name.==("Content-Length").unary_!
                  closure($anonfun)
                }
              ).groupBy[String](
                {
                  def $anonfun(_$2: sttp.model.Header): String = _$2.name
                  closure($anonfun)
                }
              ).map[String, scala.collection.immutable.::[String]](
                {
                  {
                    def $anonfun(p: (String, scala.collection.immutable.Seq[sttp.model.Header])): fx.HttpHeader = 
                      {
                        val values: scala.collection.immutable.List[String] = 
                          p._2.map[String](
                            {
                              def $anonfun(_$3: sttp.model.Header): String = _$3.value
                              closure($anonfun)
                            }
                          ).toList
                        fx.HttpHeader.apply(p._1, 
                          values.headOption.getOrElse[String](
                            <special-ops>.<cbn-arg>[("" : String)](
                              {
                                def $anonfun(): ("" : String) = ""
                                closure($anonfun)
                              }
                            )
                          )
                        , values.tail)
                      }
                    closure($anonfun)
                  }
                }
              ).toList
            )
          )
        }
      private def getResponseHeaders[T](response: java.net.http.HttpResponse[T]): fx.Http[scala.collection.immutable.Seq[sttp.model.Header]] = 
        {
          fx.Http.apply[scala.collection.immutable.Seq[sttp.model.Header]](
            jdk.CollectionConverters.MapHasAsScala[String, java.util.List[String]](response.headers().map()).asScala.toList.foldLeft[
              scala.collection.immutable.Seq[sttp.model.Header]
            ](Seq.empty[sttp.model.Header])(
              {
                {
                  def $anonfun(acc: scala.collection.immutable.Seq[sttp.model.Header], jnhHeader: (String, java.util.List[String])): 
                    scala.collection.immutable.Seq[sttp.model.Header]
                   = 
                    {
                      acc.++[sttp.model.Header](
                        jdk.CollectionConverters.ListHasAsScala[String](jnhHeader._2).asScala.map[sttp.model.Header](
                          {
                            def $anonfun(_$4: String): sttp.model.Header = sttp.model.Header.apply(jnhHeader._1, _$4)
                            closure($anonfun)
                          }
                        )
                      )
                    }
                  closure($anonfun)
                }
              }
            )
          )
        }
      private def bodyAsResponseAs[T](using x$1: sttp.monad.MonadError[fx.Http]): 
        sttp.client3.internal.BodyFromResponseAs[fx.Http, java.net.http.HttpResponse[fx.Receive[Byte]], Nothing, fx.Receive[Byte]]
       = 
        {
          final class $anon() extends 
            sttp.client3.internal.BodyFromResponseAs[fx.Http, java.net.http.HttpResponse[fx.Receive[Byte]], Nothing, fx.Receive[Byte]]
          ()(x$1) {
            protected override def withReplayableBody(response: java.net.http.HttpResponse[fx.Receive[Byte]], 
              replayableBody: scala.util.Either[Array[Byte], sttp.client3.internal.SttpFile]
            ): fx.Http[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
              fx.Http.apply[java.net.http.HttpResponse[fx.Receive[Byte]]](
                {
                  final class $anon() extends Object(), java.net.http.HttpResponse[fx.Receive[Byte]] {
                    def statusCode(): Int = response.statusCode()
                    def request(): java.net.http.HttpRequest = response.request()
                    def previousResponse(): java.util.Optional[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
                      java.util.Optional.of[java.net.http.HttpResponse[fx.Receive[Byte]]](response)
                    def headers(): java.net.http.HttpHeaders = response.headers()
                    def body(): fx.Receive[Byte] = 
                      replayableBody.fold[fx.Receive[Byte]](
                        {
                          def $anonfun(bytes: Array[Byte]): fx.Receive[Byte] = fx.streamOf[Byte](wrapByteArray(bytes).toList)
                          closure($anonfun)
                        }
                      , 
                        {
                          def $anonfun(file: sttp.client3.internal.SttpFile): fx.Receive[Byte] = 
                            fx.streamOf[Byte](wrapByteArray(file.readAsByteArray).toList)
                          closure($anonfun)
                        }
                      )
                    def sslSession(): java.util.Optional[javax.net.ssl.SSLSession] = response.sslSession()
                    def uri(): java.net.URI = response.uri()
                    def version(): java.net.http.HttpClient.Version = response.version()
                  }
                  new Object with java.net.http.HttpResponse {...}():java.net.http.HttpResponse[fx.Receive[Byte]]
                }
              )
            protected override def regularIgnore(response: java.net.http.HttpResponse[fx.Receive[Byte]]): fx.Http[Unit] = 
              fx.Http.map[fx.Receive[Byte]](
                fx.Http.apply[fx.Receive[Byte]](
                  {
                    val $proxy4: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                      fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                    {
                      def liftedTree1(): fx.Receive[Byte] = 
                        (try 
                          {
                            {
                              def $anonfun(using evidence$7: fx.Throws[Exception]): fx.Receive[Byte] = response.body()
                              closure($anonfun)
                            }.$asInstanceOf[((fx.Throws[Exception]) ?=> fx.Receive[Byte]) & (($proxy4.Throws[Exception]) ?=> fx.Receive[Byte])].apply(
                              fx.Throws.unsafeExceptions[Exception]
                            )
                          }
                         catch 
                          {
                            case r @ _:Exception => 
                              {
                                sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[fx.Receive[Byte]]
                              }
                          }
                        )
                      liftedTree1()
                    }:fx.Receive[Byte]
                  }
                )
              )[Unit](
                {
                  def $anonfun(x$1: fx.Receive[Byte]): Unit = 
                    matchResult4[Unit]: 
                      {
                        case val x19: (x$1 : fx.Receive) @unchecked = x$1:(x$1 : fx.Receive) @unchecked
                        return[matchResult4] 
                          {
                            fx.Http.apply[Unit](())
                            ()
                          }
                      }
                  closure($anonfun)
                }
              )
            protected override def regularAsByteArray(response: java.net.http.HttpResponse[fx.Receive[Byte]]): fx.Http[Array[Byte]] = 
              fx.Http.apply[Array[Byte]](
                {
                  val $proxy5: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                    fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                  {
                    def liftedTree2(): Array[Byte] = 
                      (try 
                        {
                          {
                            def $anonfun(using evidence$8: fx.Throws[Exception]): Array[Byte] = 
                              fx.toList[Byte](response.body()).toArray[Byte](scala.reflect.ClassTag.apply[Byte](classOf[Byte]))
                            closure($anonfun)
                          }.$asInstanceOf[((fx.Throws[Exception]) ?=> Array[Byte]) & (($proxy5.Throws[Exception]) ?=> Array[Byte])].apply(
                            fx.Throws.unsafeExceptions[Exception]
                          )
                        }
                       catch 
                        {
                          case r @ _:Exception => 
                            {
                              sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[Array[Byte]]
                            }
                        }
                      )
                    liftedTree2()
                  }:Array[Byte]
                }
              )
            protected override def regularAsFile(response: java.net.http.HttpResponse[fx.Receive[Byte]], file: sttp.client3.internal.SttpFile): 
              fx.Http[sttp.client3.internal.SttpFile]
             = 
              fx.Http.map[java.nio.file.Path](
                fx.Http.apply[java.nio.file.Path](
                  {
                    val $proxy6: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                      fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                    {
                      def liftedTree3(): java.nio.file.Path = 
                        (try 
                          {
                            {
                              def $anonfun(using evidence$9: fx.Throws[Exception]): java.nio.file.Path = 
                                java.nio.file.Files.write(file.toPath, 
                                  fx.toList[Byte](response.body()).toArray[Byte](scala.reflect.ClassTag.apply[Byte](classOf[Byte]))
                                , [ : java.nio.file.OpenOption])
                              closure($anonfun)
                            }.$asInstanceOf[((fx.Throws[Exception]) ?=> java.nio.file.Path) & (($proxy6.Throws[Exception]) ?=> java.nio.file.Path)].apply(
                              fx.Throws.unsafeExceptions[Exception]
                            )
                          }
                         catch 
                          {
                            case r @ _:Exception => 
                              {
                                sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[java.nio.file.Path]
                              }
                          }
                        )
                      liftedTree3()
                    }:java.nio.file.Path
                  }
                )
              )[sttp.client3.internal.SttpFile](
                {
                  def $anonfun(x$1: java.nio.file.Path): sttp.client3.internal.SttpFile = 
                    matchResult5[sttp.client3.internal.SttpFile]: 
                      {
                        case val x20: (x$1 : java.nio.file.Path) @unchecked = x$1:(x$1 : java.nio.file.Path) @unchecked
                        return[matchResult5] file
                      }
                  closure($anonfun)
                }
              )
            protected override def regularAsStream(response: java.net.http.HttpResponse[fx.Receive[Byte]]): 
              fx.Http[(fx.Receive[Byte], () => fx.Http[Unit])]
             = 
              fx.Http.apply[(fx.Receive[Byte], () => fx.Http[Unit])](
                {
                  val $proxy7: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                    fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
                  {
                    def liftedTree4(): (fx.Receive[Byte], () => fx.Http[Unit]) = 
                      (try 
                        {
                          {
                            def $anonfun(using evidence$10: fx.Throws[Exception]): (fx.Receive[Byte], () => fx.Http[Unit]) = 
                              Tuple2.apply[fx.Receive[Byte], () => fx.Http[Unit]](response.body(), 
                                {
                                  def $anonfun(): fx.Http[Unit] = fx.Http.apply[Unit](())
                                  closure($anonfun)
                                }
                              )
                            closure($anonfun)
                          }.$asInstanceOf[
                            ((fx.Throws[Exception]) ?=> (fx.Receive[Byte], () => fx.Http[Unit])) & (($proxy7.Throws[Exception]) ?=> (fx.Receive[Byte], () 
                              =>
                             fx.Http[Unit]))
                          ].apply(fx.Throws.unsafeExceptions[Exception])
                        }
                       catch 
                        {
                          case r @ _:Exception => 
                            {
                              sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(r))[(fx.Receive[Byte], () => fx.Http[Unit])]
                            }
                        }
                      )
                    liftedTree4()
                  }:(fx.Receive[Byte], () => fx.Http[Unit])
                }
              )
            protected override def handleWS[T](responseAs: sttp.client3.WebSocketResponseAs[T, ?], meta: sttp.model.ResponseMetadata, ws: Nothing): 
              fx.Http[T]
             = 
              fx.Http.apply[T](
                sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(new RuntimeException("Websockets are unsupported")))[T]
              )
            protected override def cleanupWhenNotAWebSocket(response: java.net.http.HttpResponse[fx.Receive[Byte]], 
              e: sttp.client3.ws.NotAWebSocketException
            ): fx.Http[Unit] = fx.Http.apply[Unit](())
            protected override def cleanupWhenGotWebSocket(response: Nothing, e: sttp.client3.ws.GotAWebSocketException): fx.Http[Unit] = 
              fx.Http.apply[Unit](
                sttp.fx.HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(new RuntimeException("Websockets are unsupported")))[Unit]
              )
            private val $outer: sttp.fx.HttpScalaFXBackend
            final def sttp$fx$HttpScalaFXBackend$_$$anon$$$outer: sttp.fx.HttpScalaFXBackend = $outer
          }
          new sttp.client3.internal.BodyFromResponseAs {...}():
            sttp.client3.internal.BodyFromResponseAs[fx.Http, java.net.http.HttpResponse[fx.Receive[Byte]], Nothing, fx.Receive[Byte]]
        }
      private def toResponse[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]](request: sttp.client3.Request[T, R], 
        response: java.net.http.HttpResponse[fx.Receive[Byte]]
      )(using x$3: sttp.fx.StatusCodeToStatusCode[sttp.model.StatusCode, fx.StatusCode], x$4: sttp.fx.StatusCodeToStatusCode[Int, sttp.model.StatusCode]): 
        fx.Http[sttp.client3.Response[T]]
       = 
        {
          val status: sttp.model.StatusCode = 
            fx.Http.value[sttp.model.StatusCode](x$4.toStatusCode(response.statusCode())()(HttpScalaFXBackend.this.control))(
              HttpScalaFXBackend.this.control
            )
          val statusText: String = 
            fx.StatusCode.statusText(
              fx.Http.value[fx.StatusCode](x$3.toStatusCode(status)()(HttpScalaFXBackend.this.control))(HttpScalaFXBackend.this.control)
            )
          val headers: scala.collection.immutable.Seq[sttp.model.Header] = 
            fx.Http.value[scala.collection.immutable.Seq[sttp.model.Header]](this.getResponseHeaders[fx.Receive[Byte]](response))(
              HttpScalaFXBackend.this.control
            )
          val responseMetadata: sttp.model.ResponseMetadata = sttp.model.ResponseMetadata.apply(status, statusText, headers)
          val body: T = 
            fx.Http.value[T](
              this.bodyAsResponseAs[T](this.given_MonadError_Http).apply[T](request.response, responseMetadata, 
                Left.apply[java.net.http.HttpResponse[fx.Receive[Byte]], Nothing](response)
              )
            )(HttpScalaFXBackend.this.control)
          fx.Http.apply[sttp.client3.Response[T]](sttp.client3.Response.apply[T](body, status, statusText, headers))
        }
      private def getUri[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]](request: sttp.client3.Request[T, R]): fx.Http[java.net.URI] = 
        {
          fx.Http.apply[java.net.URI](
            {
              val $proxy8: fx.Throws$package.type{Throws[R <: Exception] = Unit} = 
                fx.Throws$package.$asInstanceOf[fx.Throws$package.type{Throws[R <: Exception] = Unit}]
              {
                def liftedTree5(): java.net.URI = 
                  (try 
                    {
                      {
                        def $anonfun(using evidence$11: fx.Throws[Exception]): java.net.URI = request.uri.toJavaUri
                        closure($anonfun)
                      }.$asInstanceOf[((fx.Throws[Exception]) ?=> java.net.URI) & (($proxy8.Throws[Exception]) ?=> java.net.URI)].apply(
                        fx.Throws.unsafeExceptions[Exception]
                      )
                    }
                   catch 
                    {
                      case r @ _:Exception => 
                        {
                          def $anonfun(e: Exception): java.net.URI = HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(e))[Nothing]
                          closure($anonfun)
                        }.$asInstanceOf[Exception => java.net.URI].apply(r)
                    }
                  )
                liftedTree5()
              }:java.net.URI
            }
          )
        }
      private def makeRequest[T, R >: sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http], A](request: sttp.client3.Request[T, R], body: A)(using 
        x$3: sttp.fx.ToHttpBodyMapper[A]
      , x$4: fx.HttpResponseMapper[fx.Receive[Byte]]): fx.Http[sttp.client3.Response[T]] = 
        {
          final lazy given def bm: fx.HttpBodyMapper[A] = x$3.toHttpBodyMapper(body)()
          val headers: scala.collection.immutable.::[fx.HttpHeader] = 
            fx.Http.value[scala.collection.immutable.::[fx.HttpHeader]](
              this.getRequestHeaders[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http]), A](request, body)(bm)
            )(HttpScalaFXBackend.this.control)
          val uri: java.net.URI = 
            fx.Http.value[java.net.URI](this.getUri[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request))(
              HttpScalaFXBackend.this.control
            )
          val timeout: scala.concurrent.duration.Duration = request.options.readTimeout
          matchResult6[Function3]: 
            {
              case val x21: (request.method : (): Object) = request.method
              if sttp.model.Method.DELETE.==(x21) then 
                return[matchResult6] 
                  {
                    this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                      fx.DELETE(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                        HttpScalaFXBackend.this.control
                      , HttpScalaFXBackend.this.client)
                    )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                  }
               else ()
              if sttp.model.Method.GET.==(x21) then 
                return[matchResult6] 
                  {
                    this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                      fx.GET(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                        HttpScalaFXBackend.this.control
                      , HttpScalaFXBackend.this.client)
                    )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                  }
               else ()
              if sttp.model.Method.HEAD.==(x21) then 
                return[matchResult6] 
                  {
                    val response: fx.Http[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
                      fx.Http.map[java.net.http.HttpResponse[Void]](
                        fx.Http.apply[java.net.http.HttpResponse[Void]](
                          fx.HEAD(uri)(timeout, headers).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[Void]], HttpScalaFXBackend.this.control, 
                            HttpScalaFXBackend.this.client
                          )
                        )
                      )[java.net.http.HttpResponse[fx.Receive[Byte]]](
                        {
                          {
                            def $anonfun(response: java.net.http.HttpResponse[Void]): java.net.http.HttpResponse[fx.Receive[Byte]] = 
                              {
                                {
                                  final class $anon() extends Object(), java.net.http.HttpResponse[fx.Receive[Byte]] {
                                    def statusCode(): Int = response.statusCode()
                                    def request(): java.net.http.HttpRequest = response.request()
                                    def previousResponse(): java.util.Optional[java.net.http.HttpResponse[fx.Receive[Byte]]] = 
                                      java.util.Optional.ofNullable[java.net.http.HttpResponse[fx.Receive[Byte]]](null)
                                    def headers(): java.net.http.HttpHeaders = response.headers()
                                    def body(): fx.Receive[Byte] = fx.streamOf[Byte](wrapByteArray(Array.emptyByteArray).toList)
                                    def sslSession(): java.util.Optional[javax.net.ssl.SSLSession] = response.sslSession()
                                    def uri(): java.net.URI = response.uri()
                                    def version(): java.net.http.HttpClient.Version = response.version()
                                  }
                                  new Object with java.net.http.HttpResponse {...}():java.net.http.HttpResponse[fx.Receive[Byte]]
                                }
                              }
                            closure($anonfun)
                          }
                        }
                      )
                    this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                      fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](response)(HttpScalaFXBackend.this.control)
                    )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                  }
               else ()
              if sttp.model.Method.OPTIONS.==(x21) then 
                return[matchResult6] 
                  {
                    this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                      fx.OPTIONS(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                        HttpScalaFXBackend.this.control
                      , HttpScalaFXBackend.this.client)
                    )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                  }
               else ()
              if sttp.model.Method.PATCH.==(x21) then 
                return[matchResult6] 
                  {
                    this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                      fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](
                        sttp.fx.patch(uri)[fx.Receive[Byte]].apply[A](body, timeout, headers)(x$4, bm, HttpScalaFXBackend.this.control)
                      )(HttpScalaFXBackend.this.control)
                    )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                  }
               else ()
              if sttp.model.Method.POST.==(x21) then 
                return[matchResult6] 
                  {
                    this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                      fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](
                        sttp.fx.post(uri)[fx.Receive[Byte]].apply[A](body, timeout, headers)(x$4, bm, HttpScalaFXBackend.this.control)
                      )(HttpScalaFXBackend.this.control)
                    )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                  }
               else ()
              if sttp.model.Method.PUT.==(x21) then 
                return[matchResult6] 
                  {
                    this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                      fx.Http.value[java.net.http.HttpResponse[fx.Receive[Byte]]](
                        sttp.fx.put(uri)[fx.Receive[Byte]].apply[A](body, timeout, headers)(x$4, bm, HttpScalaFXBackend.this.control)
                      )(HttpScalaFXBackend.this.control)
                    )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                  }
               else ()
              if sttp.model.Method.TRACE.==(x21) then 
                return[matchResult6] 
                  {
                    this.toResponse[T, (sttp.fx.ReceiveStreams & sttp.capabilities.Effect[fx.Http])](request, 
                      fx.TRACE(uri)[fx.Receive[Byte]](timeout, headers)(x$4).apply(fx.defaultRetryPolicy[java.net.http.HttpResponse[fx.Receive[Byte]]], 
                        HttpScalaFXBackend.this.control
                      , HttpScalaFXBackend.this.client)
                    )(sttp.fx.StatusCodeToStatusCode.smStatusCode, sttp.fx.StatusCodeToStatusCode.given_StatusCodeToStatusCode_Int_StatusCode)
                  }
               else ()
              case val m: sttp.client3.Identity[sttp.model.Method] = x21
              return[matchResult6] 
                {
                  fx.Http.apply[sttp.client3.Response[T]](
                    HttpScalaFXBackend.this.control.shift(fx.HttpExecutionException.apply(new RuntimeException("Method: ".+(m).+(" is unsupported."))))[
                      sttp.client3.Response[T]
                    ]
                  )
                }
            }
        }
    }
  }
} of class class dotty.tools.dotc.ast.Trees$PackageDef # -1
[info] exception occurred while compiling /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/HttpExtensions.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/HttpScalaFXBackend.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/PatchPartiallyApplied.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/PostPartiallyApplied.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/PutPartiallyApplied.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/ReceiveStreams.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/StatusCodeToStatusCode.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/ToHttpBodyMapper.scala
java.lang.AssertionError: assertion failed while compiling /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/HttpExtensions.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/HttpScalaFXBackend.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/PatchPartiallyApplied.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/PostPartiallyApplied.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/PutPartiallyApplied.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/ReceiveStreams.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/StatusCodeToStatusCode.scala, /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/src/main/scala/sttp/fx/ToHttpBodyMapper.scala
[error] ## Exception when compiling 8 sources to /home/jackcviers/Documents/development/scala-fx/sttp-scala-fx/target/scala-3.1.2/classes
[error] java.lang.AssertionError: assertion failed
[error] scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:11)
[error] dotty.tools.dotc.transform.Erasure$Boxing$.adaptToType(Erasure.scala:387)
[error] dotty.tools.dotc.transform.Erasure$Typer.adapt(Erasure.scala:1042)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2937)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2941)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3057)
[error] dotty.tools.dotc.typer.Typer.$anonfun$47(Typer.scala:2256)
[error] dotty.tools.dotc.typer.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:238)
[error] dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2256)
[error] dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:932)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2786)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2871)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:121)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2937)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2941)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2963)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3013)
[error] dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1030)
[error] dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2454)
[error] dotty.tools.dotc.transform.Erasure$Typer.typedClassDef(Erasure.scala:1020)
[error] dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:2797)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2801)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2871)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:121)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2937)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2941)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2963)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3013)
[error] dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1030)
[error] dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1069)
[error] dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1073)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2817)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2872)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:121)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2937)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2941)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3057)
[error] dotty.tools.dotc.typer.Typer.$anonfun$47(Typer.scala:2256)
[error] dotty.tools.dotc.typer.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:238)
[error] dotty.tools.dotc.typer.Typer.typedDefDef(Typer.scala:2256)
[error] dotty.tools.dotc.transform.Erasure$Typer.typedDefDef(Erasure.scala:932)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2786)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2871)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:121)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2937)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2941)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2963)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3013)
[error] dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1030)
[error] dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2454)
[error] dotty.tools.dotc.transform.Erasure$Typer.typedClassDef(Erasure.scala:1020)
[error] dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:2797)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:2801)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2871)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:121)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2937)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2941)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2963)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3013)
[error] dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1030)
[error] dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2581)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2842)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2872)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:121)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2937)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2941)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:2990)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3013)
[error] dotty.tools.dotc.transform.Erasure$Typer.typedStats(Erasure.scala:1030)
[error] dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2581)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:2842)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:2872)
[error] dotty.tools.dotc.typer.ReTyper.typedUnadapted(ReTyper.scala:121)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2937)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:2941)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3057)
[error] dotty.tools.dotc.transform.Erasure.run(Erasure.scala:144)
[error] dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:308)
[error] scala.collection.immutable.List.map(List.scala:250)
[error] dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:309)
[error] dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:259)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1328)
[error] dotty.tools.dotc.Run.runPhases$1(Run.scala:270)
[error] dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:278)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
[error] dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:68)
[error] dotty.tools.dotc.Run.compileUnits(Run.scala:287)
[error] dotty.tools.dotc.Run.compileSources(Run.scala:220)
[error] dotty.tools.dotc.Run.compile(Run.scala:204)
[error] dotty.tools.dotc.Driver.doCompile(Driver.scala:39)
[error] dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error] dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:192)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:247)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:182)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:163)
[error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:163)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:210)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:528)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:528)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:177)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:175)
[error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:461)
[error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:416)
[error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:503)
[error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:403)
[error] sbt.internal.inc.Incremental$.apply(Incremental.scala:169)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:528)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:482)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:420)
[error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2366)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2316)
[error] sbt.internal.server.BspCompileTask$.$anonfun$compute$1(BspCompileTask.scala:30)
[error] sbt.internal.io.Retry$.apply(Retry.scala:46)
[error] sbt.internal.io.Retry$.apply(Retry.scala:28)
[error] sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:30)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2314)
[error] scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error] sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error] sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error] sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error] sbt.Execute.work(Execute.scala:291)
[error] sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error] sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[error] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[error] java.base/java.lang.Thread.run(Thread.java:1589)
[error]            
[error] stack trace is suppressed; run last sttp-scala-fx / Compile / compileIncremental for the full output
[error] (sttp-scala-fx / Compile / compileIncremental) java.lang.AssertionError: assertion failed
[error] Total time: 2 s, completed Sep 8, 2022, 6:23:26 PM
sbt:root> 
```
There's precedent for this type of [compiler bug before](https://github.com/lampepfl/dotty/issues/11466). Typing Http as `type Http[A]` doesn't result in the compiler bug.

Reply:
As I'm fairly certain they're well-typed since the alias works, I think I'll leave it.